// terra-form-field/lib/terra-dev-site/test/form-field/FieldInteractiveInvalid.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Field = _interopRequireDefault(require("../../../Field"));

var _FormFieldTestCommonModule = _interopRequireDefault(require("./FormFieldTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var cx = _bind.default.bind(_FormFieldTestCommonModule.default);

var FieldExamples = /*#__PURE__*/function (_React$Component) {
  _inherits(FieldExamples, _React$Component);

  function FieldExamples(props) {
    var _this;

    _classCallCheck(this, FieldExamples);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(FieldExamples).call(this, props));
    _this.state = {
      isInvalid: false
    };
    _this.handleIsInvalidChange = _this.handleIsInvalidChange.bind(_assertThisInitialized(_this));
    return _this;
  }

  _createClass(FieldExamples, [{
    key: "handleIsInvalidChange",
    value: function handleIsInvalidChange() {
      this.setState(function (prevState) {
        return {
          isInvalid: !prevState.isInvalid
        };
      });
    }
  }, {
    key: "render",
    value: function render() {
      var field = _react.default.createElement(_Field.default, {
        id: "default",
        className: cx('form-field'),
        label: "Field Label",
        help: "Help Message",
        error: "Error Message",
        isInvalid: this.state.isInvalid
      }, _react.default.createElement("div", {
        className: cx('field-content')
      }, "Control Placeholder"));

      var optional = _react.default.createElement(_Field.default, {
        id: "optional",
        className: cx('form-field'),
        label: "Field Label",
        help: "Help Message",
        error: "Error Message",
        isInvalid: this.state.isInvalid,
        showOptional: true
      }, _react.default.createElement("div", {
        className: cx('field-content')
      }, "Control Placeholder"));

      var required = _react.default.createElement(_Field.default, {
        id: "required",
        className: cx('form-field'),
        label: "Field Label",
        help: "Help Message",
        error: "Error Message",
        isInvalid: this.state.isInvalid,
        required: true
      }, _react.default.createElement("div", {
        className: cx('field-content')
      }, "Control Placeholder"));

      var requiredHidden = _react.default.createElement(_Field.default, {
        id: "required-hidden",
        className: cx('form-field'),
        label: "Field Label",
        help: "Help Message",
        error: "Error Message",
        isInvalid: this.state.isInvalid,
        required: true,
        hideRequired: true
      }, _react.default.createElement("div", {
        className: cx('field-content')
      }, "Control Placeholder"));

      var hiddenLabel = _react.default.createElement(_Field.default, {
        id: "label-hidden",
        className: cx('form-field'),
        label: "Field Label",
        help: "Help Message",
        error: "Error Message",
        isInvalid: this.state.isInvalid,
        isLabelHidden: true
      }, _react.default.createElement("div", {
        className: cx('field-content')
      }, "Control Placeholder"));

      var inline = _react.default.createElement("div", null, _react.default.createElement(_Field.default, {
        id: "inline-1",
        className: cx('form-field'),
        label: "Field Label - Spacing POC",
        error: "Error Message",
        isInline: true,
        isInvalid: this.state.isInvalid,
        required: true
      }, _react.default.createElement("div", {
        className: cx('field-content')
      }, "Control Placeholder")), _react.default.createElement(_Field.default, {
        id: "inline-2",
        className: cx('form-field'),
        label: "Field Label - Spacing POC",
        error: "Error Message",
        isInline: true,
        isInvalid: this.state.isInvalid,
        required: true,
        hideRequired: true
      }, _react.default.createElement("div", {
        className: cx('field-content')
      }, "Control Placeholder")), _react.default.createElement(_Field.default, {
        id: "inline-3",
        className: cx('form-field'),
        label: "Field Label - Spacing POC",
        error: "Error Message",
        isInline: true,
        isInvalid: this.state.isInvalid,
        showOptional: true
      }, _react.default.createElement("div", {
        className: cx('field-content')
      }, "Control Placeholder")));

      return _react.default.createElement("div", null, _react.default.createElement("button", {
        type: "button",
        id: "toggle-is-invalid",
        onClick: this.handleIsInvalidChange
      }, "Toggle In Error "), _react.default.createElement("h3", null, "Field"), field, _react.default.createElement("h3", null, "Field - Optional"), optional, _react.default.createElement("h3", null, "Field - Required"), required, _react.default.createElement("h3", null, "Field - Required but Indicator Hidden"), requiredHidden, _react.default.createElement("h3", null, "Field - Hidden Label"), hiddenLabel, _react.default.createElement("h3", null, "Field - Inline"), inline);
    }
  }]);

  return FieldExamples;
}(_react.default.Component);

var _default = FieldExamples;
exports.default = _default;
