// terra-form-field/lib/terra-dev-site/test/form-field/FieldInline.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Field = _interopRequireDefault(require("../../../Field"));

var _FormFieldTestCommonModule = _interopRequireDefault(require("./FormFieldTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_FormFieldTestCommonModule.default);

var FieldExamples = function FieldExamples() {
  return _react.default.createElement("div", null, _react.default.createElement(_Field.default, {
    className: cx('form-field'),
    label: "Field Label",
    isInline: true,
    required: true
  }, _react.default.createElement("div", {
    className: cx('field-content')
  }, "Control Placeholder")), _react.default.createElement(_Field.default, {
    className: cx('form-field'),
    label: "Field Label",
    isInline: true,
    showOptional: true
  }, _react.default.createElement("div", {
    className: cx('field-content')
  }, "Control Placeholder")), _react.default.createElement(_Field.default, {
    className: cx('form-field'),
    label: "Field Label",
    isInline: true,
    required: true
  }, _react.default.createElement("div", {
    className: cx('field-content')
  }, "Control Placeholder")), _react.default.createElement("br", null), _react.default.createElement(_Field.default, {
    className: cx('form-field'),
    label: "Field Label",
    isInline: true,
    error: "Error Message",
    isInvalid: true,
    required: true
  }, _react.default.createElement("div", {
    className: cx('field-content')
  }, "Control Placeholder")), _react.default.createElement(_Field.default, {
    className: cx('form-field'),
    label: "Field Label",
    isInline: true,
    error: "Error Message",
    isInvalid: true,
    showOptional: true
  }, _react.default.createElement("div", {
    className: cx('field-content')
  }, "Control Placeholder")), _react.default.createElement(_Field.default, {
    className: cx('form-field'),
    label: "Field Label",
    isInline: true,
    error: "Error Message",
    isInvalid: true,
    required: true
  }, _react.default.createElement("div", {
    className: cx('field-content')
  }, "Control Placeholder")));
};

var _default = FieldExamples;
exports.default = _default;
