// terra-form-field/lib/terra-dev-site/test/form-field/FieldErrorText.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Field = _interopRequireDefault(require("../../../Field"));

var _FormFieldTestCommonModule = _interopRequireDefault(require("./FormFieldTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_FormFieldTestCommonModule.default);

var FieldErrorText = function FieldErrorText() {
  return _react.default.createElement("div", null, _react.default.createElement("h3", null, "Field - Error Text"), _react.default.createElement(_Field.default, {
    id: "error-text",
    className: cx('form-field'),
    label: "Field Label",
    isLabelHidden: true,
    error: "Error Message",
    isInvalid: true
  }, _react.default.createElement("div", {
    className: cx('field-content')
  }, "Control Placeholder")));
};

var _default = FieldErrorText;
exports.default = _default;
