// terra-slide-panel-manager/lib/terra-dev-site/doc/slide-panel-manager/SlidePanelManager.1.doc.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraDocTemplate = _interopRequireDefault(require("terra-doc-template"));

var _package = require("../../../../package.json");

var _README = _interopRequireDefault(require("../../../../docs/README.md"));

var _SlidePanelManager = _interopRequireDefault(require("!raw-loader!../../../../src/SlidePanelManager.jsx"));

var _SlidePanelManagerExample = _interopRequireDefault(require("../example/SlidePanelManagerExample"));

var _SlidePanelManagerExample2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/SlidePanelManagerExample.jsx"));

// Component Source
// Examples
var DocPage = function DocPage() {
  return _react.default.createElement(_terraDocTemplate.default, {
    packageName: _package.name,
    readme: _README.default,
    srcPath: "https://github.com/cerner/terra-framework/tree/master/packages/".concat(_package.name),
    examples: [{
      title: 'Overlay Example',
      example: _react.default.createElement(_SlidePanelManagerExample.default, {
        behavior: "overlay-1"
      }),
      source: _SlidePanelManagerExample2.default
    }, {
      title: 'Squish Example',
      example: _react.default.createElement(_SlidePanelManagerExample.default, {
        behavior: "squish-2"
      }),
      source: _SlidePanelManagerExample2.default
    }, {
      title: 'Disclosure Accessory Example',
      example: _react.default.createElement(_SlidePanelManagerExample.default, {
        behavior: "squish-3",
        showAccessory: true
      }),
      source: _SlidePanelManagerExample2.default
    }],
    propsTables: [{
      componentSrc: _SlidePanelManager.default,
      source: _SlidePanelManagerExample2.default
    }]
  });
};

var _default = DocPage;
exports.default = _default;
