// terra-search-field/lib/SearchField.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _terraButton = _interopRequireDefault(require("terra-button"));

var KeyCode = _interopRequireWildcard(require("keycode-js"));

var _IconSearch = _interopRequireDefault(require("terra-icon/lib/icon/IconSearch"));

var _terraFormInput = _interopRequireDefault(require("terra-form-input"));

var _reactIntl = require("react-intl");

var _SearchFieldModule = _interopRequireDefault(require("./SearchField.module.scss"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var cx = _bind.default.bind(_SearchFieldModule.default);

var Icon = _react.default.createElement(_IconSearch.default, null);

var propTypes = {
  /**
   * The defaultValue of the search field. Use this to create an uncontrolled search field.
   */
  defaultValue: _propTypes.default.string,

  /**
   * When true, will disable the auto-search.
   */
  disableAutoSearch: _propTypes.default.bool,

  /**
   * Callback ref to pass into the inner input component.
   */
  inputRefCallback: _propTypes.default.func,

  /**
   * Custom input attributes to apply to the input field such as aria-label.
   */
  // eslint-disable-next-line react/forbid-prop-types
  inputAttributes: _propTypes.default.object,

  /**
   * @private
   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
   */
  intl: _reactIntl.intlShape.isRequired,

  /**
   * Whether or not the field should display as a block.
   */
  isBlock: _propTypes.default.bool,

  /**
   * When true, will disable the field.
   */
  isDisabled: _propTypes.default.bool,

  /**
   * The minimum number of characters to perform a search.
   */
  minimumSearchTextLength: _propTypes.default.number,

  /**
   * Function to trigger when user changes the input value. Provide a function to create a controlled input.
   */
  onChange: _propTypes.default.func,

  /**
   * Function to trigger when user inputs a value. Use when programmatically setting a value. Sends parameter {Event} event.
   */
  onInput: _propTypes.default.func,

  /**
   * A callback to indicate an invalid search. Sends parameter {String} searchText.
   */
  onInvalidSearch: _propTypes.default.func,

  /**
   * A callback to perform search. Sends parameter {String} searchText.
   */
  onSearch: _propTypes.default.func,

  /**
   * Placeholder text to show while the search field is empty.
   */
  placeholder: _propTypes.default.string,

  /**
   * How long the component should wait (in milliseconds) after input before performing an automatic search.
   */
  searchDelay: _propTypes.default.number,

  /**
   * The value of search field.  Use this to create a controlled search field.
   */
  value: _propTypes.default.string
};
var defaultProps = {
  defaultValue: undefined,
  disableAutoSearch: false,
  isBlock: false,
  isDisabled: false,
  minimumSearchTextLength: 2,
  placeholder: '',
  searchDelay: 250,
  value: undefined,
  inputAttributes: undefined
};

var SearchField = /*#__PURE__*/function (_React$Component) {
  _inherits(SearchField, _React$Component);

  function SearchField(props) {
    var _this;

    _classCallCheck(this, SearchField);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(SearchField).call(this, props));
    _this.handleClear = _this.handleClear.bind(_assertThisInitialized(_this));
    _this.handleTextChange = _this.handleTextChange.bind(_assertThisInitialized(_this));
    _this.handleSearch = _this.handleSearch.bind(_assertThisInitialized(_this));
    _this.handleKeyDown = _this.handleKeyDown.bind(_assertThisInitialized(_this));
    _this.handleInput = _this.handleInput.bind(_assertThisInitialized(_this));
    _this.setInputRef = _this.setInputRef.bind(_assertThisInitialized(_this));
    _this.updateSearchText = _this.updateSearchText.bind(_assertThisInitialized(_this));
    _this.searchTimeout = null;
    _this.searchText = _this.props.defaultValue || _this.props.value;
    return _this;
  }

  _createClass(SearchField, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate() {
      // if consumer updates the value prop with onChange, need to update variable to match
      this.updateSearchText(this.props.value);
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      this.clearSearchTimeout();
    }
  }, {
    key: "setInputRef",
    value: function setInputRef(node) {
      this.inputRef = node;

      if (this.props.inputRefCallback) {
        this.props.inputRefCallback(node);
      }
    }
  }, {
    key: "handleClear",
    value: function handleClear(event) {
      // Pass along changes to consuming components using associated props
      if (this.props.onChange) {
        this.props.onChange(event, '');
      }

      if (this.props.onInvalidSearch) {
        this.props.onInvalidSearch('');
      }

      this.updateSearchText(''); // Clear input field

      if (this.inputRef) {
        this.inputRef.value = '';
        this.inputRef.focus();
      }
    }
  }, {
    key: "handleTextChange",
    value: function handleTextChange(event) {
      var textValue = event.target.value;
      this.updateSearchText(textValue);

      if (this.props.onChange) {
        this.props.onChange(event, textValue);
      }

      if (!this.searchTimeout && !this.props.disableAutoSearch) {
        this.searchTimeout = setTimeout(this.handleSearch, this.props.searchDelay);
      }
    }
  }, {
    key: "updateSearchText",
    value: function updateSearchText(searchText) {
      if (typeof searchText !== 'undefined' && searchText !== this.searchText) {
        this.searchText = searchText; // Forcing update for clearButton rerender.

        this.forceUpdate();
      }
    }
  }, {
    key: "handleInput",
    value: function handleInput(event) {
      var textValue = event.target.value;
      this.updateSearchText(textValue);

      if (this.props.onInput) {
        this.props.onInput(event);
      }
    }
  }, {
    key: "handleKeyDown",
    value: function handleKeyDown(event) {
      if (event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
        this.handleSearch();
      }

      if (event.nativeEvent.keyCode === KeyCode.KEY_ESCAPE) {
        this.handleClear(event);
      }
    }
  }, {
    key: "handleSearch",
    value: function handleSearch() {
      this.clearSearchTimeout();
      var searchText = this.searchText || '';

      if (searchText.length >= this.props.minimumSearchTextLength && this.props.onSearch) {
        this.props.onSearch(searchText);
      } else if (this.props.onInvalidSearch) {
        this.props.onInvalidSearch(searchText);
      }
    }
  }, {
    key: "clearSearchTimeout",
    value: function clearSearchTimeout() {
      if (this.searchTimeout) {
        clearTimeout(this.searchTimeout);
        this.searchTimeout = null;
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          defaultValue = _this$props.defaultValue,
          disableAutoSearch = _this$props.disableAutoSearch,
          inputRefCallback = _this$props.inputRefCallback,
          inputAttributes = _this$props.inputAttributes,
          intl = _this$props.intl,
          isBlock = _this$props.isBlock,
          isDisabled = _this$props.isDisabled,
          minimumSearchTextLength = _this$props.minimumSearchTextLength,
          onChange = _this$props.onChange,
          onInput = _this$props.onInput,
          onInvalidSearch = _this$props.onInvalidSearch,
          onSearch = _this$props.onSearch,
          placeholder = _this$props.placeholder,
          searchDelay = _this$props.searchDelay,
          value = _this$props.value,
          customProps = _objectWithoutProperties(_this$props, ["defaultValue", "disableAutoSearch", "inputRefCallback", "inputAttributes", "intl", "isBlock", "isDisabled", "minimumSearchTextLength", "onChange", "onInput", "onInvalidSearch", "onSearch", "placeholder", "searchDelay", "value"]);

      var searchFieldClassNames = cx(['search-field', {
        block: isBlock
      }, customProps.className]);
      var inputText = inputAttributes && Object.prototype.hasOwnProperty.call(inputAttributes, 'aria-label') ? inputAttributes['aria-label'] : intl.formatMessage({
        id: 'Terra.searchField.search'
      });
      var buttonText = intl.formatMessage({
        id: 'Terra.searchField.submit-search'
      });
      var clearText = intl.formatMessage({
        id: 'Terra.searchField.clear'
      });

      var additionalInputAttributes = _objectSpread({}, inputAttributes);

      var clearIcon = _react.default.createElement("span", {
        className: cx('clear-icon')
      });

      if (value !== undefined) {
        additionalInputAttributes.value = value;
      } else {
        additionalInputAttributes.defaultValue = defaultValue;
      }

      var clearButton = this.searchText && !isDisabled ? _react.default.createElement(_terraButton.default, {
        className: cx('clear'),
        onClick: this.handleClear,
        text: clearText,
        variant: "utility",
        icon: clearIcon,
        isIconOnly: true
      }) : undefined;
      return _react.default.createElement("div", _extends({}, customProps, {
        className: searchFieldClassNames
      }), _react.default.createElement("div", {
        className: cx('input-group')
      }, _react.default.createElement(_terraFormInput.default, _extends({
        className: cx('input'),
        type: "search",
        placeholder: placeholder,
        onChange: this.handleTextChange,
        disabled: isDisabled,
        ariaLabel: inputText,
        "aria-disabled": isDisabled,
        onKeyDown: this.handleKeyDown,
        onInput: this.handleInput,
        refCallback: this.setInputRef
      }, additionalInputAttributes)), clearButton), _react.default.createElement(_terraButton.default, {
        className: cx('button'),
        text: buttonText,
        onClick: this.handleSearch,
        isDisabled: isDisabled,
        icon: Icon,
        isIconOnly: true,
        isCompact: true
      }));
    }
  }]);

  return SearchField;
}(_react.default.Component);

SearchField.propTypes = propTypes;
SearchField.defaultProps = defaultProps;

var _default = (0, _reactIntl.injectIntl)(SearchField);

exports.default = _default;
