// terra-overlay/lib/terra-dev-site/test/overlay/LoadingOverlay/ClearLoadingOverlay.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _OverlayContainer = _interopRequireDefault(require("../../../../OverlayContainer"));

var _LoadingOverlay = _interopRequireDefault(require("../../../../LoadingOverlay"));

var _OverlayTestCommonModule = _interopRequireDefault(require("../Overlay/OverlayTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_OverlayTestCommonModule.default);

var _default = function _default() {
  return _react.default.createElement(_OverlayContainer.default, {
    className: cx('overlay-container1'),
    overlay: _react.default.createElement(_LoadingOverlay.default, {
      isOpen: true,
      backgroundStyle: _LoadingOverlay.default.Opts.BackgroundStyles.CLEAR,
      isRelativeToContainer: true,
      id: "terra-LoadingOverlay"
    })
  });
};

exports.default = _default;
