// terra-menu/lib/terra-dev-site/test/menu/menu-item/MenuItemWithInstructionsForUse.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _terraList = _interopRequireDefault(require("terra-list"));

var _Menu = _interopRequireDefault(require("../../../../Menu"));

var childContextTypes = {
  isSelectableMenu: _propTypes.default.bool
};

var MenuItemWithInstructionsForUse = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(MenuItemWithInstructionsForUse, _React$Component);

  function MenuItemWithInstructionsForUse() {
    (0, _classCallCheck2.default)(this, MenuItemWithInstructionsForUse);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(MenuItemWithInstructionsForUse).apply(this, arguments));
  }

  (0, _createClass2.default)(MenuItemWithInstructionsForUse, [{
    key: "getChildContext",
    value: function getChildContext() {
      return {
        isSelectableMenu: true
      };
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement("div", {
        id: "isSelected"
      }, _react.default.createElement("p", null, "Menu Items with eIFU Icon")), _react.default.createElement(_terraList.default, {
        role: "menu"
      }, _react.default.createElement(_Menu.default.Item, {
        text: "eIFU Icon Menu Item",
        key: "1",
        className: "TestMenuItem",
        isInstructionsForUse: true
      }), _react.default.createElement(_Menu.default.Item, {
        isSelectable: true,
        text: "eIFU Icon Nested Menu",
        key: "2",
        className: "TestMenuItem",
        isInstructionsForUse: true,
        subMenuItems: [_react.default.createElement(_Menu.default.Item, {
          text: "Default 2.1",
          key: "2.1",
          className: "TestNestedMenuContent"
        })]
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Selectable+Selected eIFU Icon Menu Item",
        key: "3",
        className: "TestMenuItem",
        isSelectable: true,
        isSelected: true,
        isInstructionsForUse: true
      })));
    }
  }]);
  return MenuItemWithInstructionsForUse;
}(_react.default.Component);

MenuItemWithInstructionsForUse.childContextTypes = childContextTypes;
var _default = MenuItemWithInstructionsForUse;
exports.default = _default;
