// terra-menu/lib/terra-dev-site/test/menu/menu-item/MenuItemOnClick.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _Menu = _interopRequireDefault(require("../../../../Menu"));

var MenuItemOnClick = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(MenuItemOnClick, _React$Component);

  function MenuItemOnClick(props) {
    var _this;

    (0, _classCallCheck2.default)(this, MenuItemOnClick);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(MenuItemOnClick).call(this, props));
    _this.state = {
      clickNumber: 0
    };
    _this.handleSelection = _this.handleSelection.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(MenuItemOnClick, [{
    key: "handleSelection",
    value: function handleSelection() {
      this.setState(function (prevState) {
        return {
          clickNumber: prevState.clickNumber + 1
        };
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement("div", {
        id: "clickNumber"
      }, _react.default.createElement("h3", null, "Item has been clicked", this.state.clickNumber, ' ', "times")), _react.default.createElement("ul", {
        role: "menu"
      }, _react.default.createElement(_Menu.default.Item, {
        text: "OnClick Menu Item",
        key: "1",
        className: "TestOnClickItem",
        onClick: this.handleSelection
      })));
    }
  }]);
  return MenuItemOnClick;
}(_react.default.Component);

var _default = MenuItemOnClick;
exports.default = _default;
