// terra-menu/lib/terra-dev-site/test/menu/menu-item/MenuItemDisabled.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _terraList = _interopRequireDefault(require("terra-list"));

var _Menu = _interopRequireDefault(require("../../../../Menu"));

var childContextTypes = {
  isSelectableMenu: _propTypes.default.bool
};

var MenuItemDisabled = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(MenuItemDisabled, _React$Component);

  function MenuItemDisabled() {
    (0, _classCallCheck2.default)(this, MenuItemDisabled);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(MenuItemDisabled).apply(this, arguments));
  }

  (0, _createClass2.default)(MenuItemDisabled, [{
    key: "getChildContext",
    value: function getChildContext() {
      return {
        isSelectableMenu: true
      };
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement("div", {
        id: "isSelected"
      }, _react.default.createElement("p", null, "The item is disabled. The selection state should not change with any user interaction.")), _react.default.createElement(_terraList.default, {
        role: "menu"
      }, _react.default.createElement(_Menu.default.Item, {
        text: "Disabled Menu Item",
        key: "1",
        className: "TestDisabledItem",
        isSelectable: true,
        isDisabled: true
      }), _react.default.createElement(_Menu.default.Item, {
        isDisabled: true,
        isSelectable: true,
        text: "Nested Menu",
        key: "2",
        className: "TestNestedMenu",
        subMenuItems: [_react.default.createElement(_Menu.default.Item, {
          text: "Default 2.1",
          key: "2.1",
          className: "TestNestedMenuContent"
        })]
      }), _react.default.createElement(_Menu.default.Item, {
        isDisabled: true,
        text: "Disabled Selected Menu Item",
        key: "3",
        className: "TestDisabledItem",
        isSelectable: true,
        isSelected: true
      }), _react.default.createElement(_Menu.default.Item, {
        isDisabled: true,
        text: "Disabled eIFU Icon Menu Item",
        key: "4",
        className: "TestDisabledItem",
        isInstructionsForUse: true
      })));
    }
  }]);
  return MenuItemDisabled;
}(_react.default.Component);

MenuItemDisabled.childContextTypes = childContextTypes;
var _default = MenuItemDisabled;
exports.default = _default;
