// terra-application-navigation/lib/drawer-menu/_DrawerMenuUser.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _terraAvatar = _interopRequireDefault(require("terra-avatar"));

var _propTypes2 = require("../utils/propTypes");

var _DrawerMenuUserModule = _interopRequireDefault(require("./DrawerMenuUser.module.scss"));

var cx = _bind.default.bind(_DrawerMenuUserModule.default);

var propTypes = {
  /**
   * A configuration object with information pertaining to the application's user.
   */
  userConfig: _propTypes2.userConfigPropType.isRequired,

  /**
   * Size variant of the user. One of values [`small`, `large`].
   */
  variant: _propTypes.default.oneOf(['small', 'large'])
};
var defaultProps = {
  variant: 'small'
};

var DrawerMenuUser = function DrawerMenuUser(_ref) {
  var userConfig = _ref.userConfig,
      variant = _ref.variant;
  return _react.default.createElement("div", {
    className: variant === 'small' ? cx('small-user-layout') : cx('large-user-layout')
  }, _react.default.createElement("div", {
    className: cx('avatar-container')
  }, _react.default.createElement("div", {
    className: cx('avatar-outline')
  }), _react.default.createElement("div", {
    className: cx('avatar-inner')
  }, _react.default.createElement(_terraAvatar.default, {
    alt: userConfig.name,
    image: userConfig.imageSrc,
    initials: userConfig.initials
  }))), _react.default.createElement("div", {
    className: cx('info-container')
  }, _react.default.createElement("div", {
    "aria-hidden": true,
    className: cx('name')
  }, userConfig.name), userConfig.detail ? _react.default.createElement("div", {
    className: cx('detail')
  }, userConfig.detail) : null));
};

DrawerMenuUser.propTypes = propTypes;
DrawerMenuUser.defaultProps = defaultProps;
var _default = DrawerMenuUser;
exports.default = _default;
