// terra-application/lib/terra-dev-site/doc/application/demo/LoadingOverlayPresenter.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _applicationLoadingOverlay = _interopRequireDefault(require("terra-application/lib/application-loading-overlay"));

var LoadingOverlayPresenter = function LoadingOverlayPresenter() {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      showLoadingOverlay = _useState2[0],
      setShowLoadingOverlay = _useState2[1];

  var timeoutRef = (0, _react.useRef)();
  (0, _react.useEffect)(function () {
    return function () {
      window.clearTimeout(timeoutRef.current);
    };
  }, []);
  return _react.default.createElement("div", null, _react.default.createElement("h3", null, "Loading Overlay Presenter"), _react.default.createElement("p", null, "This component uses the ApplicationLoadingOverlay to present a loading overlay over itself. The overlay in this demo is shown for 2 seconds."), _react.default.createElement("p", null, _react.default.createElement("button", {
    disabled: showLoadingOverlay,
    type: "button",
    onClick: function onClick() {
      setShowLoadingOverlay(true);
      timeoutRef.current = setTimeout(function () {
        setShowLoadingOverlay(false);
      }, 2000);
    }
  }, "Show Loading Overlay")), _react.default.createElement(_applicationLoadingOverlay.default, {
    isOpen: showLoadingOverlay,
    backgroundStyle: "light"
  }));
};

var _default = LoadingOverlayPresenter;
exports.default = _default;
