// terra-abstract-modal/lib/terra-dev-site/doc/example/AbstractModalIsFullscreen.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _terraAbstractModal = _interopRequireDefault(require("terra-abstract-modal"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _AbstractModalDocCommonModule = _interopRequireDefault(require("./AbstractModalDocCommon.module.scss"));

var cx = _bind.default.bind(_AbstractModalDocCommonModule.default);

var AbstractModalIsFullscreen = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(AbstractModalIsFullscreen, _React$Component);

  function AbstractModalIsFullscreen() {
    var _this;

    (0, _classCallCheck2.default)(this, AbstractModalIsFullscreen);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(AbstractModalIsFullscreen).call(this));
    _this.state = {
      isOpen: false
    };
    _this.handleOpenModal = _this.handleOpenModal.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleCloseModal = _this.handleCloseModal.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(AbstractModalIsFullscreen, [{
    key: "handleOpenModal",
    value: function handleOpenModal() {
      this.setState({
        isOpen: true
      });
    }
  }, {
    key: "handleCloseModal",
    value: function handleCloseModal() {
      this.setState({
        isOpen: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement(_terraAbstractModal.default, {
        ariaLabel: "Fullscreen Modal",
        isOpen: this.state.isOpen,
        isFullscreen: true,
        onRequestClose: this.handleCloseModal
      }, _react.default.createElement("div", {
        className: cx('content-wrapper')
      }, _react.default.createElement("h1", null, "Fullscreen Modal"), _react.default.createElement("br", null), _react.default.createElement("p", null, "This modal will always take up the full screen."), _react.default.createElement("p", null), _react.default.createElement("br", null), _react.default.createElement("button", {
        type: "button",
        onClick: this.handleCloseModal
      }, "Close Modal"))), _react.default.createElement("button", {
        type: "button",
        onClick: this.handleOpenModal
      }, "Open Modal"));
    }
  }]);
  return AbstractModalIsFullscreen;
}(_react.default.Component);

var _default = AbstractModalIsFullscreen;
exports.default = _default;
