// terra-abstract-modal/lib/terra-dev-site/doc/example/AbstractModalAlertDialog.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _terraAbstractModal = _interopRequireDefault(require("terra-abstract-modal"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _ExampleAbstractSizeModule = _interopRequireDefault(require("./ExampleAbstractSize.module.scss"));

var _AbstractModalDocCommonModule = _interopRequireDefault(require("./AbstractModalDocCommon.module.scss"));

var cx = _bind.default.bind(_AbstractModalDocCommonModule.default);
/* eslint-disable-next-line react/prop-types */


var AlertDialogContent = function AlertDialogContent(_ref) {
  var handleCloseModal = _ref.handleCloseModal;
  var confirmBtn = (0, _react.useRef)(null);
  (0, _react.useEffect)(function () {
    confirmBtn.current.focus();
  }, []);
  return _react.default.createElement("div", {
    id: "alert-dialog-content",
    className: cx('content-wrapper')
  }, _react.default.createElement("h1", null, "Alert Dialog Modal"), _react.default.createElement("br", null), _react.default.createElement("p", null, "The abstract modal can be used to create an alert dialog modal. You can use the role,", ' ', _react.default.createElement("code", null, "alertdialog"), ' ', ", to create a modal dialog that interrupts the users workflow to get a response, usually some sort of confirmation."), _react.default.createElement("button", {
    ref: confirmBtn,
    type: "button",
    onClick: handleCloseModal
  }, "Confirm action"), _react.default.createElement("button", {
    type: "button",
    onClick: handleCloseModal
  }, "Close Modal"));
};
/* VoiceOver will not read the content of the modal dialog if the role attribute
 is set to `alertdialog` and ariaLabel is defined */


var ariaLabel = '';

var AbstractModalAlertDialog = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(AbstractModalAlertDialog, _React$Component);

  function AbstractModalAlertDialog() {
    var _this;

    (0, _classCallCheck2.default)(this, AbstractModalAlertDialog);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(AbstractModalAlertDialog).call(this));
    _this.state = {
      isOpen: false
    };
    _this.handleOpenModal = _this.handleOpenModal.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleCloseModal = _this.handleCloseModal.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(AbstractModalAlertDialog, [{
    key: "handleOpenModal",
    value: function handleOpenModal() {
      this.setState({
        isOpen: true
      });
    }
  }, {
    key: "handleCloseModal",
    value: function handleCloseModal() {
      this.setState({
        isOpen: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement(_terraAbstractModal.default, {
        ariaLabel: ariaLabel,
        "aria-labelledby": "alert-dialog-content",
        isOpen: this.state.isOpen,
        onRequestClose: this.handleCloseModal,
        closeOnOutsideClick: false,
        classNameModal: _ExampleAbstractSizeModule.default['fixed-size'],
        role: "alertdialog"
      }, _react.default.createElement(AlertDialogContent, {
        handleCloseModal: this.handleCloseModal
      })), _react.default.createElement("button", {
        type: "button",
        onClick: this.handleOpenModal
      }, "Open Modal"));
    }
  }]);
  return AbstractModalAlertDialog;
}(_react.default.Component);

var _default = AbstractModalAlertDialog;
exports.default = _default;
