// terra-application-navigation/lib/terra-dev-site/test/application-navigation/HeroAndUser.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _applicationBase = _interopRequireDefault(require("terra-application/lib/application-base"));

var _ApplicationNavigation = _interopRequireDefault(require("../../../ApplicationNavigation"));

var _Placeholder = _interopRequireDefault(require("terra-doc-template/lib/Placeholder"));

/* eslint-disable */
var hero = _react.default.createElement(_Placeholder.default, {
  title: "Test Hero",
  variant: "light"
});

var userConfig = {
  name: 'Test Name',
  detail: 'Test Detail'
}; // TODO: remove terra-application after it is incorporated into dev-site for themes or themes are co-located

var HeroAndUserTest = function HeroAndUserTest() {
  return _react.default.createElement(_applicationBase.default, {
    locale: "en"
  }, _react.default.createElement(_ApplicationNavigation.default, {
    hero: hero,
    userConfig: userConfig
  }));
};

var _default = HeroAndUserTest;
exports.default = _default;
