// terra-application-navigation/lib/terra-dev-site/test/application-navigation/User.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _applicationBase = _interopRequireDefault(require("terra-application/lib/application-base"));

var _ApplicationNavigation = _interopRequireDefault(require("../../../ApplicationNavigation"));

var _FixedHeight = _interopRequireDefault(require("../common/FixedHeight"));

/* eslint-disable */
var userConfig = {
  name: 'Test Name'
};
var userConfigDetail = {
  name: 'Test Name',
  detail: 'Test Detail'
};
var userConfigInitials = {
  name: 'Test Name',
  initials: 'TN'
};
var userConfigSrc = {
  name: 'Test Name',
  imageSrc: 'test-src'
}; // TODO: remove terra-application after it is incorporated into dev-site for themes or themes are co-located

var UserTest = function UserTest() {
  return _react.default.createElement(_applicationBase.default, {
    locale: "en"
  }, _react.default.createElement(_FixedHeight.default, null, _react.default.createElement(_ApplicationNavigation.default, {
    userConfig: userConfig
  })), _react.default.createElement(_FixedHeight.default, null, _react.default.createElement(_ApplicationNavigation.default, {
    userConfig: userConfigDetail
  })), _react.default.createElement(_FixedHeight.default, null, _react.default.createElement(_ApplicationNavigation.default, {
    userConfig: userConfigInitials
  })), _react.default.createElement(_FixedHeight.default, null, _react.default.createElement(_ApplicationNavigation.default, {
    userConfig: userConfigSrc
  })));
};

var _default = UserTest;
exports.default = _default;
