// terra-application-navigation/lib/terra-dev-site/test/application-navigation/Utilities.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _applicationBase = _interopRequireDefault(require("terra-application/lib/application-base"));

var _IconSearch = _interopRequireDefault(require("terra-icon/lib/icon/IconSearch"));

var _IconPill = _interopRequireDefault(require("terra-icon/lib/icon/IconPill"));

var _IconVisualization = _interopRequireDefault(require("terra-icon/lib/icon/IconVisualization"));

var _IconLightbulb = _interopRequireDefault(require("terra-icon/lib/icon/IconLightbulb"));

var _ApplicationNavigation = _interopRequireDefault(require("../../../ApplicationNavigation"));

var utilityItems = [{
  icon: _react.default.createElement(_IconSearch.default, null),
  key: 'item-a',
  text: 'Item A',
  metaData: {
    test: 'a'
  }
}, {
  icon: _react.default.createElement(_IconPill.default, null),
  key: 'item-b',
  text: 'Item B',
  metaData: {
    test: 'b'
  }
}, {
  icon: _react.default.createElement(_IconVisualization.default, null),
  key: 'item-c',
  text: 'Item C',
  metaData: {
    test: 'c'
  }
}, {
  icon: _react.default.createElement(_IconLightbulb.default, null),
  key: 'item-d',
  text: 'Item D',
  metaData: {
    test: 'd'
  }
}];
/* eslint-disable no-alert */

var handleItemSelection = function handleItemSelection(key, metaData) {
  alert("itemKey: ".concat(key, ", metaData: ").concat(metaData.test));
};
/* eslint-enable no-alert */
// TODO: remove terra-application after it is incorporated into dev-site for themes or themes are co-located


var UtilitiesTest = function UtilitiesTest() {
  return _react.default.createElement(_applicationBase.default, {
    locale: "en"
  }, _react.default.createElement(_ApplicationNavigation.default, {
    onSelectSettings: function onSelectSettings() {
      return alert('Settings Selected');
    } // eslint-disable-line no-alert
    ,
    onSelectHelp: function onSelectHelp() {
      return alert('Help Selected');
    } // eslint-disable-line no-alert
    ,
    onSelectLogout: function onSelectLogout() {
      return alert('Logout Selected');
    } // eslint-disable-line no-alert
    ,
    utilityItems: utilityItems,
    onSelectUtilityItem: handleItemSelection
  }));
};

var _default = UtilitiesTest;
exports.default = _default;
