// terra-slide-group/lib/terra-dev-site/doc/example/SlideGroupDemoAnimated.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _terraSlideGroup = _interopRequireDefault(require("terra-slide-group"));

var _SlideGroupDemoModule = _interopRequireDefault(require("./SlideGroupDemo.module.scss"));

var cx = _bind.default.bind(_SlideGroupDemoModule.default);

var SlideGroupDemo = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(SlideGroupDemo, _React$Component);

  function SlideGroupDemo(props) {
    var _this;

    (0, _classCallCheck2.default)(this, SlideGroupDemo);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(SlideGroupDemo).call(this, props));
    _this.increment = _this.increment.bind((0, _assertThisInitialized2.default)(_this));
    _this.decrement = _this.decrement.bind((0, _assertThisInitialized2.default)(_this));
    _this.state = {
      counter: 1
    };
    return _this;
  }

  (0, _createClass2.default)(SlideGroupDemo, [{
    key: "increment",
    value: function increment() {
      this.setState(function (prevState) {
        return {
          counter: prevState.counter + 1
        };
      });
    }
  }, {
    key: "decrement",
    value: function decrement() {
      this.setState(function (prevState) {
        return {
          counter: prevState.counter - 1
        };
      });
    }
  }, {
    key: "render",
    value: function render() {
      var slides = [];

      for (var i = 0; i < this.state.counter; i += 1) {
        slides.push(_react.default.createElement("div", {
          key: "Slide ".concat(i),
          className: cx('slide')
        }, _react.default.createElement("h2", null, "Slide", i), _react.default.createElement("br", null), _react.default.createElement("button", {
          type: "button",
          onClick: this.increment
        }, "Increment"), i !== 0 ? _react.default.createElement("button", {
          type: "button",
          onClick: this.decrement
        }, "Decrement") : null));
      }

      return _react.default.createElement("div", {
        className: cx('container')
      }, _react.default.createElement(_terraSlideGroup.default, {
        items: slides,
        isAnimated: this.props.isAnimated
      }));
    }
  }]);
  return SlideGroupDemo;
}(_react.default.Component);

SlideGroupDemo.propTypes = {
  isAnimated: _propTypes.default.bool
};

var AnimatedSlideGroup = function AnimatedSlideGroup() {
  return _react.default.createElement(SlideGroupDemo, {
    isAnimated: true
  });
};

var _default = AnimatedSlideGroup;
exports.default = _default;
