// terra-doc-template/lib/terra-dev-site/doc/doc-template/DocTemplate.1.doc.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _DocTemplate = _interopRequireDefault(require("../../../../lib/DocTemplate"));

var _README = _interopRequireDefault(require("../../../../docs/README.md"));

var _package = require("../../../../package.json");

var _DocTemplate2 = _interopRequireDefault(require("!terra-props-table-loader!../../../../src/DocTemplate"));

var _DocTemplate3 = _interopRequireDefault(require("!raw-loader!../../../../src/DocTemplate"));

var _BadgeExamples = _interopRequireDefault(require("../example/BadgeExamples"));

var _BadgeExamples2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/BadgeExamples"));

var _DocTemplate4 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/doc-template/DocTemplate.1.doc"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable import/no-webpack-loader-syntax, import/first, import/no-duplicates */

/* eslint-disable-next-line import/no-self-import */
var DocTemplateExamples = function DocTemplateExamples() {
  return _react.default.createElement(_DocTemplate.default, {
    packageName: _package.name,
    packageVersion: _package.version,
    readme: _README.default,
    srcPath: "https://github.com/cerner/terra-core/tree/master/packages/terra-doc-template",
    examples: [{
      title: 'DocTemplate',
      source: _DocTemplate4.default,
      example: _react.default.createElement("p", null, "This very page was generated using the DocTemplate. The template supports more than one Props Table and example, even though those were not featured here.")
    }, {
      title: 'Package Badge Examples',
      source: _BadgeExamples2.default,
      example: _react.default.createElement(_BadgeExamples.default, null)
    }],
    propsTables: [{
      componentName: 'Doc Template',
      componentSrc: _DocTemplate3.default,
      componentProps: _DocTemplate2.default
    }]
  });
};

var _default = DocTemplateExamples;
exports.default = _default;
