// terra-doc-template/lib/terra-dev-site/doc/doc-template/Placeholder.2.doc.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _DocTemplate = _interopRequireDefault(require("../../../DocTemplate"));

var _PLACEHOLDER = _interopRequireDefault(require("../../../../docs/PLACEHOLDER.md"));

var _package = require("../../../../package.json");

var _Placeholder = _interopRequireDefault(require("!terra-props-table-loader!../../../../src/Placeholder"));

var _Placeholder2 = _interopRequireDefault(require("!raw-loader!../../../../src/Placeholder"));

var _PlaceholderDefault = _interopRequireDefault(require("../example/PlaceholderDefault"));

var _PlaceholderDefault2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/PlaceholderDefault"));

var _PlaceholderLight = _interopRequireDefault(require("../example/PlaceholderLight"));

var _PlaceholderLight2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/doc/example/PlaceholderLight"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// Example Files
var DocPage = function DocPage() {
  return _react.default.createElement(_DocTemplate.default, {
    packageName: _package.name,
    packageVersion: _package.version,
    readme: _PLACEHOLDER.default,
    srcPath: "https://github.com/cerner/terra-core/tree/master/packages/".concat(_package.name),
    examples: [{
      title: 'Dark Placeholder',
      example: _react.default.createElement(_PlaceholderDefault.default, null),
      source: _PlaceholderDefault2.default
    }, {
      title: 'Light Placeholder',
      example: _react.default.createElement(_PlaceholderLight.default, null),
      source: _PlaceholderLight2.default
    }],
    propsTables: [{
      componentName: 'Placeholder',
      componentSrc: _Placeholder2.default,
      componentProps: _Placeholder.default
    }]
  });
};

var _default = DocPage;
exports.default = _default;
