// terra-popup/lib/terra-dev-site/test/popup/ArrowSmallTargetRightPopup.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Popup = _interopRequireDefault(require("../../../Popup"));

var _ArrowSmallTargetRightPopupTestModule = _interopRequireDefault(require("./ArrowSmallTargetRightPopup.test.module.scss"));

var cx = _bind.default.bind(_ArrowSmallTargetRightPopupTestModule.default); // This tests verifies the PopupUtils.getContentOffset methed when (targetNode.clientWidth < segment) & attachment is right


var OffsetPopup = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(OffsetPopup, _React$Component);

  function OffsetPopup(props) {
    var _this;

    (0, _classCallCheck2.default)(this, OffsetPopup);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(OffsetPopup).call(this, props));
    _this.handleButtonClick = _this.handleButtonClick.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleRequestClose = _this.handleRequestClose.bind((0, _assertThisInitialized2.default)(_this));
    _this.setButtonNode = _this.setButtonNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.getButtonNode = _this.getButtonNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.setParentNode = _this.setParentNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.getParentNode = _this.getParentNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.state = {
      open: true
    };
    return _this;
  }

  (0, _createClass2.default)(OffsetPopup, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.forceUpdate();
    }
  }, {
    key: "setButtonNode",
    value: function setButtonNode(node) {
      this.buttonNode = node;
    }
  }, {
    key: "getButtonNode",
    value: function getButtonNode() {
      return this.buttonNode;
    }
  }, {
    key: "setParentNode",
    value: function setParentNode(node) {
      this.parentNode = node;
    }
  }, {
    key: "getParentNode",
    value: function getParentNode() {
      return this.parentNode;
    }
  }, {
    key: "handleButtonClick",
    value: function handleButtonClick() {
      this.setState({
        open: true
      });
    }
  }, {
    key: "handleRequestClose",
    value: function handleRequestClose() {
      this.setState({
        open: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", {
        id: "test-popup-area",
        className: cx('test-popup-area'),
        ref: this.setParentNode
      }, _react.default.createElement(_Popup.default, {
        boundingRef: this.getParentNode,
        classNameArrow: "test-arrow",
        classNameContent: "test-content",
        contentAttachment: "top right",
        contentHeight: "80",
        contentWidth: "160",
        isArrowDisplayed: true,
        isOpen: this.state.open,
        onRequestClose: this.handleRequestClose,
        targetRef: this.getButtonNode
      }, _react.default.createElement("p", {
        className: cx('popup-text')
      }, "This popup was adjusted because the target was smaller than the arrow position allowed.")), _react.default.createElement("button", {
        type: "button",
        id: "offset-button",
        className: cx('popup-button'),
        onClick: this.handleButtonClick,
        ref: this.setButtonNode
      }, ''));
    }
  }]);
  return OffsetPopup;
}(_react.default.Component);

var _default = OffsetPopup;
exports.default = _default;
