// terra-infinite-list/lib/terra-dev-site/test/infinite-list/InfiniteListVirtual.test.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _index = _interopRequireWildcard(require("../../../index"));

var _InfiniteListVirtualTestModule = _interopRequireDefault(require("./InfiniteListVirtual.test.module.scss"));

var cx = _bind.default.bind(_InfiniteListVirtualTestModule.default);

var InfiniteListUpdating = function InfiniteListUpdating() {
  return _react.default.createElement("div", null, _react.default.createElement("button", {
    type: "button",
    id: "test-click",
    onClick: function onClick() {
      document.querySelector('#test-infinite-list').scrollTop = 1600;
    }
  }, "button"), _react.default.createElement("div", {
    className: cx('infinite-list-wrapper')
  }, _react.default.createElement(_index.default, {
    id: "test-infinite-list",
    dividerStyle: "standard",
    isFinishedLoading: true,
    ariaLabel: "Virtual"
  }, _react.default.createElement(_index.Item, {
    key: "item-0"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 0")), _react.default.createElement(_index.Item, {
    key: "item-1"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 1")), _react.default.createElement(_index.Item, {
    key: "item-2"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 2")), _react.default.createElement(_index.Item, {
    key: "item-3"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 3")), _react.default.createElement(_index.Item, {
    key: "item-4"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 4")), _react.default.createElement(_index.Item, {
    key: "item-5"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 5")), _react.default.createElement(_index.Item, {
    key: "item-6"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 6")), _react.default.createElement(_index.Item, {
    key: "item-7"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 7")), _react.default.createElement(_index.Item, {
    key: "item-8"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 8")), _react.default.createElement(_index.Item, {
    key: "item-9"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 9")), _react.default.createElement(_index.Item, {
    key: "item-10"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 10")), _react.default.createElement(_index.Item, {
    key: "item-11"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 11")), _react.default.createElement(_index.Item, {
    key: "item-12"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 12")), _react.default.createElement(_index.Item, {
    key: "item-13"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 13")), _react.default.createElement(_index.Item, {
    key: "item-14"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 14")), _react.default.createElement(_index.Item, {
    key: "item-15"
  }, _react.default.createElement("div", {
    className: cx('item-content')
  }, "Item 15")))));
};

var _default = InfiniteListUpdating;
exports.default = _default;
