// terra-infinite-list/lib/terra-dev-site/test/infinite-list/InfiniteListUpdating.test.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _index = _interopRequireWildcard(require("../../../index"));

var _InfiniteListTestCommonModule = _interopRequireDefault(require("./InfiniteListTestCommon.module.scss"));

var cx = _bind.default.bind(_InfiniteListTestCommonModule.default);

var InfiniteListUpdating = function InfiniteListUpdating() {
  return _react.default.createElement(_index.default, {
    id: "test-infinite-list",
    isFinishedLoading: false,
    ariaLabel: "Updating",
    progressiveLoadingIndicator: _react.default.createElement("div", {
      className: cx('progressive-loading-indicator')
    }, "Progressive Loading")
  }, _react.default.createElement(_index.Item, {
    key: "item-0"
  }, _react.default.createElement("div", {
    className: cx('item-type1')
  }, "Item 0")), _react.default.createElement(_index.Item, {
    key: "item-1"
  }, _react.default.createElement("div", {
    className: cx('item-type2')
  }, "Item 1")), _react.default.createElement(_index.Item, {
    key: "item-2"
  }, _react.default.createElement("div", {
    className: cx('item-type1')
  }, "Item 2")), _react.default.createElement(_index.Item, {
    key: "item-3"
  }, _react.default.createElement("div", {
    className: cx('item-type2')
  }, "Item 3")));
};

var _default = InfiniteListUpdating;
exports.default = _default;
