// terra-form-select/lib/terra-dev-site/doc/example/search/Incomplete.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SearchSelect = _interopRequireDefault(require("terra-form-select/lib/SearchSelect"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _FormSelectDocCommonModule = _interopRequireDefault(require("../FormSelectDocCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_FormSelectDocCommonModule.default);

var IncompleteExample = function IncompleteExample() {
  return _react.default.createElement(_SearchSelect.default, {
    placeholder: "Select a color",
    required: true,
    isIncomplete: true,
    className: cx('form-select')
  }, _react.default.createElement(_SearchSelect.default.Option, {
    value: "blue",
    display: "Blue"
  }), _react.default.createElement(_SearchSelect.default.Option, {
    value: "green",
    display: "Green"
  }), _react.default.createElement(_SearchSelect.default.Option, {
    value: "purple",
    display: "Purple"
  }), _react.default.createElement(_SearchSelect.default.Option, {
    value: "red",
    display: "Red"
  }), _react.default.createElement(_SearchSelect.default.Option, {
    value: "violet",
    display: "Violet"
  }));
};

var _default = IncompleteExample;
exports.default = _default;
