// terra-form-select/lib/terra-dev-site/doc/example/search/RequiredSearchSelectField.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SearchSelectField = _interopRequireDefault(require("terra-form-select/lib/SearchSelectField"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _FormSelectDocCommonModule = _interopRequireDefault(require("../FormSelectDocCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_FormSelectDocCommonModule.default);

var SearchSelectFieldExample = function SearchSelectFieldExample() {
  return _react.default.createElement(_SearchSelectField.default, {
    label: "T-shirt size",
    placeholder: "Select a size",
    selectId: "tshirt-size-field-2",
    required: true,
    className: cx('form-select')
  }, _react.default.createElement(_SearchSelectField.default.Option, {
    value: "xSmall",
    display: "Extra Small"
  }), _react.default.createElement(_SearchSelectField.default.Option, {
    value: "small",
    display: "Small"
  }), _react.default.createElement(_SearchSelectField.default.Option, {
    value: "medium",
    display: "Medium"
  }), _react.default.createElement(_SearchSelectField.default.Option, {
    value: "large",
    display: "Large"
  }), _react.default.createElement(_SearchSelectField.default.Option, {
    value: "xLarge",
    display: "Extra Large"
  }));
};

var _default = SearchSelectFieldExample;
exports.default = _default;
