// terra-spacer/lib/terra-dev-site/doc/example/SpacerExample.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraButton = _interopRequireDefault(require("terra-button"));

var _terraSpacer = _interopRequireDefault(require("terra-spacer"));

require("terra-spacer/lib/terra-dev-site/doc/example/Spacer.module.scss");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SpacerExample = function SpacerExample() {
  return _react.default.createElement("div", null, _react.default.createElement(_terraSpacer.default, {
    className: "spacerdemodefault",
    paddingTop: "large+4",
    paddingBottom: "large+4",
    paddingLeft: "large+4",
    paddingRight: "large+4",
    isInlineBlock: true
  }, _react.default.createElement(_terraButton.default, {
    text: "Default"
  })), _react.default.createElement(_terraSpacer.default, {
    className: "spacerdemoprimary",
    paddingTop: "large+2",
    paddingBottom: "large+2",
    paddingLeft: "large+2",
    paddingRight: "large+2",
    isInlineBlock: true
  }, _react.default.createElement(_terraButton.default, {
    text: "Emphasis",
    variant: "emphasis"
  })));
};

var _default = SpacerExample;
exports.default = _default;
