// terra-spacer/lib/terra-dev-site/doc/example/SpacerShortVsLongHandExample.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraButton = _interopRequireDefault(require("terra-button"));

var _terraSpacer = _interopRequireDefault(require("terra-spacer"));

require("terra-spacer/lib/terra-dev-site/doc/example/Spacer.module.scss");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SpacerShortVsLongHandExample = function SpacerShortVsLongHandExample() {
  return _react.default.createElement("div", null, _react.default.createElement(_terraSpacer.default, {
    className: "spacerdemoprimary",
    padding: "large small",
    margin: "medium large+1",
    isInlineBlock: true
  }, _react.default.createElement(_terraButton.default, {
    text: "Shorthand"
  })), _react.default.createElement(_terraSpacer.default, {
    className: "spacerdemodefault",
    paddingTop: "large",
    paddingBottom: "large",
    paddingLeft: "small",
    paddingRight: "small",
    marginTop: "medium",
    marginBottom: "medium",
    marginLeft: "large+1",
    marginRight: "large+1",
    isInlineBlock: true
  }, _react.default.createElement(_terraButton.default, {
    text: "Longhand"
  })));
};

var _default = SpacerShortVsLongHandExample;
exports.default = _default;
