// terra-hyperlink/lib/terra-dev-site/test/hyperlink/ScaledHyperlink.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Hyperlink = _interopRequireDefault(require("../../../Hyperlink"));

var _HyperlinkTestCommonModule = _interopRequireDefault(require("./HyperlinkTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_HyperlinkTestCommonModule.default);

var ScaledHyperlink = function ScaledHyperlink() {
  return _react.default.createElement("div", {
    role: "main",
    className: cx('scaled-hyperlink'),
    id: "link-example"
  }, _react.default.createElement(_Hyperlink.default, {
    href: "https://www.cerner.com",
    variant: "audio"
  }, "Audio hyperlink"), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://www.cerner.com",
    variant: "document"
  }, "Document hyperlink"), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://www.cerner.com",
    variant: "external"
  }, "External hyperlink"), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://www.cerner.com",
    variant: "image"
  }, "Image hyperlink"), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://www.cerner.com",
    variant: "video"
  }, "Video hyperlink"));
};

var _default = ScaledHyperlink;
exports.default = _default;
