// terra-hyperlink/lib/terra-dev-site/test/hyperlink/StatesHyperlink.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Hyperlink = _interopRequireDefault(require("../../../Hyperlink"));

var _HyperlinkTestCommonModule = _interopRequireDefault(require("./HyperlinkTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_HyperlinkTestCommonModule.default);

var HyperlinkStates = function HyperlinkStates() {
  return _react.default.createElement("div", {
    role: "main",
    className: cx('hyperlink-states'),
    id: "link-example"
  }, _react.default.createElement(_Hyperlink.default, {
    href: "https://my.cerner.com/?"
  }, "hyperlink"), _react.default.createElement("br", null), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://my.cerner.com/?",
    className: "is-visited"
  }, "visited hyperlink"), _react.default.createElement("br", null), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://my.cerner.com/?",
    className: "is-hovered"
  }, "hovered hyperlink"), _react.default.createElement("br", null), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://my.cerner.com/?",
    className: "is-focused"
  }, "focused hyperlink"), _react.default.createElement("br", null), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://my.cerner.com/?",
    className: "is-active"
  }, "active hyperlink"), _react.default.createElement("br", null), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://my.cerner.com/?",
    variant: "document"
  }, "icon hyperlink"), _react.default.createElement("br", null), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://my.cerner.com/?",
    className: "is-visited",
    variant: "document"
  }, "icon visited hyperlink"), _react.default.createElement("br", null), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://my.cerner.com/?",
    className: "is-hovered",
    variant: "document"
  }, "icon hovered hyperlink"), _react.default.createElement("br", null), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://my.cerner.com/?",
    className: "is-focused",
    variant: "document"
  }, "icon focused hyperlink"), _react.default.createElement("br", null), _react.default.createElement("br", null), _react.default.createElement(_Hyperlink.default, {
    href: "https://my.cerner.com/?",
    className: "is-active",
    variant: "document"
  }, "icon active hyperlink"));
};

var _default = HyperlinkStates;
exports.default = _default;
