// terra-image/lib/terra-dev-site/test/image/ImageNonFluid.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Image = _interopRequireDefault(require("../../../Image"));

var _x = _interopRequireDefault(require("../../assets/150x150.jpg"));

var _ImageTestCommonModule = _interopRequireDefault(require("./ImageTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ImageTestCommonModule.default);

var ImageNonFluid = function ImageNonFluid() {
  return _react.default.createElement("div", null, _react.default.createElement("h2", null, "Non Fluid Examples"), _react.default.createElement("p", null, "Image width and height are smaller than container."), _react.default.createElement("div", {
    id: "smaller",
    className: cx('image-conatiner')
  }, _react.default.createElement(_Image.default, {
    src: _x.default,
    alt: "rounded",
    variant: "rounded"
  })), _react.default.createElement("p", null, "Image width is larger than the container. Image height is smaller than the container."), _react.default.createElement("div", {
    id: "height-smaller",
    className: cx('image-conatiner-reduced-width')
  }, _react.default.createElement(_Image.default, {
    src: _x.default,
    alt: "rounded",
    variant: "rounded"
  })), _react.default.createElement("p", null, "Image width is smaller than the container. Image height is larger than the container."), _react.default.createElement("div", {
    id: "width-smaller",
    className: cx('image-conatiner-reduced-height')
  }, _react.default.createElement(_Image.default, {
    src: _x.default,
    alt: "rounded",
    variant: "rounded"
  })));
};

var _default = ImageNonFluid;
exports.default = _default;
