// terra-notification-dialog/lib/terra-dev-site/doc/example/CompleteNotificationDialogWithLongMessage.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _terraButton = _interopRequireDefault(require("terra-button"));

var _terraNotificationDialog = _interopRequireWildcard(require("terra-notification-dialog"));

var clickConfirm = function clickConfirm() {
  console.log('You clicked confirm'); // eslint-disable-line no-console
};

var CompleteNotificationDialogWithLongMessage = function CompleteNotificationDialogWithLongMessage() {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      isOpen = _useState2[0],
      setIsOpen = _useState2[1];

  var handleOpenModal = function handleOpenModal() {
    setIsOpen(true);
  };

  var handleCloseModal = function handleCloseModal() {
    setIsOpen(false);
  };

  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_terraNotificationDialog.default, {
    variant: _terraNotificationDialog.NotificationDialogVariants.SUCCESS,
    isOpen: isOpen,
    title: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. In ornare odio quis turpis viverra, volutpat laoreet magna porta.",
    startMessage: "Quisque egestas ullamcorper velit vitae volutpat. Quisque vestibulum nulla nunc, eget pharetra massa semper ac. In sit amet felis tincidunt, laoreet tortor nec, tempus ipsum.",
    acceptAction: {
      text: 'Confirm',
      onClick: clickConfirm
    },
    rejectAction: {
      text: 'Close',
      onClick: handleCloseModal
    },
    buttonOrder: "acceptFirst",
    emphasizedAction: "accept"
  }), _react.default.createElement(_terraButton.default, {
    text: "Trigger Notification Dialog",
    onClick: handleOpenModal
  }));
};

var _default = CompleteNotificationDialogWithLongMessage;
exports.default = _default;
