// terra-notification-dialog/lib/terra-dev-site/doc/example/ContentNotificationDialog.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _terraButton = _interopRequireDefault(require("terra-button"));

var _NotificationDialog = _interopRequireWildcard(require("../../../NotificationDialog"));

var clickConfirm = function clickConfirm() {
  console.log('You clicked confirm'); // eslint-disable-line no-console
};

var ContentNotificationDialog = function ContentNotificationDialog() {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      isOpen = _useState2[0],
      setIsOpen = _useState2[1];

  var handleOpenModal = function handleOpenModal() {
    setIsOpen(true);
  };

  var handleCloseModal = function handleCloseModal() {
    setIsOpen(false);
  };

  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_NotificationDialog.default, {
    variant: _NotificationDialog.NotificationDialogVariants.ALERT,
    isOpen: isOpen,
    title: "Make sure that the title relates directly to the choices.",
    startMessage: "The Start Message is text used to provide more detail or define terminology at the beginning of the dialog body. Don\u2019t repeat the title verbatim.",
    content: _react.default.createElement("ul", null, _react.default.createElement("li", null, "Item 1"), _react.default.createElement("li", null, "Item 2")),
    endMessage: "The End Message is text used to provide more detail or define terminology at the end of the dialog body. Don\u2019t repeat the title verbatim.",
    acceptAction: {
      text: 'Confirm',
      onClick: clickConfirm
    },
    rejectAction: {
      text: 'Close',
      onClick: handleCloseModal
    },
    buttonOrder: "acceptFirst",
    emphasizedAction: "accept"
  }), _react.default.createElement(_terraButton.default, {
    text: "Trigger NotificationDialog",
    onClick: handleOpenModal
  }));
};

var _default = ContentNotificationDialog;
exports.default = _default;
