// terra-notification-dialog/lib/terra-dev-site/doc/example/NotificationDialogModalContainer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _terraButton = _interopRequireDefault(require("terra-button"));

var _terraDisclosureManager = require("terra-disclosure-manager");

var _NotificationDialogWithFocus = _interopRequireDefault(require("./NotificationDialogWithFocus"));

var NotificationDialogModalContainer = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(NotificationDialogModalContainer, _React$Component);

  function NotificationDialogModalContainer(props) {
    var _this;

    (0, _classCallCheck2.default)(this, NotificationDialogModalContainer);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(NotificationDialogModalContainer).call(this, props));
    _this.disclose = _this.disclose.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(NotificationDialogModalContainer, [{
    key: "disclose",
    value: function disclose() {
      this.props.disclosureManager.disclose({
        preferredType: 'modal',
        content: {
          key: 'DemoContainer',
          component: _react.default.createElement(_NotificationDialogWithFocus.default, null)
        }
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_terraButton.default, {
        text: "Disclose",
        onClick: this.disclose
      }));
    }
  }]);
  return NotificationDialogModalContainer;
}(_react.default.Component);

NotificationDialogModalContainer.propTypes = {
  disclosureManager: _terraDisclosureManager.disclosureManagerShape
};

var _default = (0, _terraDisclosureManager.withDisclosureManager)(NotificationDialogModalContainer);

exports.default = _default;
