// terra-popup/lib/terra-dev-site/doc/example/PopupBounded.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _terraButton = _interopRequireDefault(require("terra-button"));

var _terraPopup = _interopRequireDefault(require("terra-popup"));

var _Placeholder = _interopRequireDefault(require("terra-doc-template/lib/Placeholder"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _PopupDocCommonModule = _interopRequireDefault(require("./PopupDocCommon.module.scss"));

var cx = _bind.default.bind(_PopupDocCommonModule.default);

function PopupBounded() {
  var buttonElement = (0, _react.useRef)();
  var parentElement = (0, _react.useRef)();

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      open = _useState2[0],
      setOpen = _useState2[1];

  var setParentNode = function setParentNode(node) {
    parentElement.current = node;
  };

  var getParentNode = function getParentNode() {
    return parentElement.current;
  };

  var setButtonNode = function setButtonNode(node) {
    buttonElement.current = node;
  };

  var getButtonNode = function getButtonNode() {
    return buttonElement.current;
  };

  var handleButtonClick = function handleButtonClick() {
    setOpen(true);
  };

  var handleRequestClose = function handleRequestClose() {
    setOpen(false);
  };

  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("div", {
    className: cx('content-wrapper'),
    ref: setParentNode
  }, _react.default.createElement(_terraButton.default, {
    text: "Bounded Popup",
    onClick: handleButtonClick,
    refCallback: setButtonNode
  }), _react.default.createElement(_terraPopup.default, {
    boundingRef: getParentNode,
    classNameArrow: "test-arrow",
    classNameContent: "test-content",
    contentHeight: "240",
    contentWidth: "320",
    isOpen: open,
    onRequestClose: handleRequestClose,
    targetRef: getButtonNode,
    isContentFocusDisabled: true
  }, _react.default.createElement(_Placeholder.default, {
    title: "Popup Content"
  }))));
}

var _default = PopupBounded;
exports.default = _default;
