// terra-infinite-list/lib/terra-dev-site/doc/example/ProgressiveLoadingExample.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraInfiniteList = _interopRequireWildcard(require("terra-infinite-list"));

var _Placeholder = _interopRequireDefault(require("terra-doc-template/lib/Placeholder"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _MyProgressiveLoadingIndicator = _interopRequireDefault(require("../common/MyProgressiveLoadingIndicator"));

var _MyExampleContainer = _interopRequireDefault(require("../common/MyExampleContainer"));

var _InfiniteListDocExampleCommonModule = _interopRequireDefault(require("./InfiniteListDocExampleCommon.module.scss"));

var cx = _bind.default.bind(_InfiniteListDocExampleCommonModule.default);

var ProgressiveLoadingExample = function ProgressiveLoadingExample() {
  return _react.default.createElement(_MyExampleContainer.default, null, _react.default.createElement(_terraInfiniteList.default, {
    dividerStyle: "standard",
    ariaLabel: "Progressive Loading",
    progressiveLoadingIndicator: _react.default.createElement(_MyProgressiveLoadingIndicator.default, null)
  }, _react.default.createElement(_terraInfiniteList.Item, {
    key: "item-1"
  }, _react.default.createElement(_Placeholder.default, {
    title: "Item 1",
    className: cx('placeholder')
  }))));
};

var _default = ProgressiveLoadingExample;
exports.default = _default;
