// terra-infinite-list/lib/terra-dev-site/doc/example/VirtualizedExample.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraInfiniteList = _interopRequireWildcard(require("terra-infinite-list"));

var _Placeholder = _interopRequireDefault(require("terra-doc-template/lib/Placeholder"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _virtualizedData = _interopRequireDefault(require("./mock-data/virtualizedData"));

var _MyExampleContainer = _interopRequireDefault(require("../common/MyExampleContainer"));

var _InfiniteListDocExampleCommonModule = _interopRequireDefault(require("./InfiniteListDocExampleCommon.module.scss"));

var cx = _bind.default.bind(_InfiniteListDocExampleCommonModule.default);

var createListItem = function createListItem(itemData) {
  return _react.default.createElement(_terraInfiniteList.Item, {
    key: itemData.key
  }, _react.default.createElement(_Placeholder.default, {
    title: itemData.title,
    className: cx('placeholder')
  }));
};

var createItems = function createItems(data) {
  return data.map(function (item) {
    return createListItem(item);
  });
};

var VirtualizedExample = function VirtualizedExample() {
  return _react.default.createElement(_MyExampleContainer.default, null, _react.default.createElement(_terraInfiniteList.default, {
    dividerStyle: "standard",
    isFinishedLoading: true,
    ariaLabel: "Virtualized"
  }, createItems(_virtualizedData.default)));
};

var _default = VirtualizedExample;
exports.default = _default;
