// terra-application-navigation/lib/tabs/_Tabs.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _resizeObserverPolyfill = _interopRequireDefault(require("resize-observer-polyfill"));

var _lodash = _interopRequireDefault(require("lodash.debounce"));

var _reactIntl = require("react-intl");

var _terraPopup = _interopRequireDefault(require("terra-popup"));

var _Tab = _interopRequireDefault(require("./_Tab"));

var _TabRollup = _interopRequireDefault(require("./_TabRollup"));

var _PopupMenu = _interopRequireDefault(require("../common/_PopupMenu"));

var _propTypes2 = require("../utils/propTypes");

var _TabsModule = _interopRequireDefault(require("./Tabs.module.scss"));

var cx = _bind.default.bind(_TabsModule.default);

var propTypes = {
  /**
   * An array of configuration objects with information specifying the creation of navigation items.
   */
  navigationItems: _propTypes2.navigationItemsPropType,

  /**
   * A function to be executed for the render of each navigation item.
   */
  navigationRenderFunction: _propTypes.default.func,

  /**
   * A string identifying the currently active tab.
   */
  activeTabKey: _propTypes.default.string,

  /**
   * A function to be executed upon selection of a tab.
   */
  onTabSelect: _propTypes.default.func,

  /**
   * Key/Value pairs associating a string key entry to a numerical notification count.
   */
  notifications: _propTypes.default.object,

  /**
   * @private
   * Object containing intl APIs.
   */
  intl: _reactIntl.intlShape.isRequired
};
var defaultProps = {
  navigationItems: [],
  notifications: {}
};

var Tabs = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(Tabs, _React$Component);

  function Tabs(props) {
    var _this;

    (0, _classCallCheck2.default)(this, Tabs);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Tabs).call(this, props));
    _this.getRollupTabWidth = _this.getRollupTabWidth.bind((0, _assertThisInitialized2.default)(_this));
    _this.closePopup = _this.closePopup.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleResize = _this.handleResize.bind((0, _assertThisInitialized2.default)(_this));
    _this.renderRollup = _this.renderRollup.bind((0, _assertThisInitialized2.default)(_this));
    _this.renderPopup = _this.renderPopup.bind((0, _assertThisInitialized2.default)(_this));
    _this.buildVisibleChildren = _this.buildVisibleChildren.bind((0, _assertThisInitialized2.default)(_this));
    _this.updateSize = (0, _lodash.default)(_this.updateSize.bind((0, _assertThisInitialized2.default)(_this)), 100);
    _this.initializeResize = _this.initializeResize.bind((0, _assertThisInitialized2.default)(_this));
    _this.removeResize = _this.removeResize.bind((0, _assertThisInitialized2.default)(_this));
    _this.containerRef = _react.default.createRef();
    _this.rollupTabRef = _react.default.createRef();
    _this.rollupInnerRef = _react.default.createRef();
    _this.childRefs = [];
    _this.previousNotifications = null;
    _this.resizeListenerAdded = false;

    _this.resetCalculations();

    _this.state = {
      popupIsOpen: false
    };
    return _this;
  }

  (0, _createClass2.default)(Tabs, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      if (this.props.navigationItems && this.props.navigationItems.length) {
        this.initializeResize();
      } else {
        this.removeResize();
      }
    }
  }, {
    key: "shouldComponentUpdate",
    value: function shouldComponentUpdate(nextProps) {
      var _this$props = this.props,
          navigationItems = _this$props.navigationItems,
          activeTabKey = _this$props.activeTabKey,
          notifications = _this$props.notifications;

      if (navigationItems.length !== nextProps.navigationItems.length || activeTabKey !== nextProps.activeTabKey) {
        this.resetCalculations();
      }

      this.previousNotifications = notifications;
      return true;
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this$props2 = this.props,
          activeTabKey = _this$props2.activeTabKey,
          navigationItems = _this$props2.navigationItems;
      var popupIsOpen = this.state.popupIsOpen;

      if (navigationItems && navigationItems.length) {
        this.initializeResize();

        if (this.isCalculating) {
          this.isCalculating = false;
          this.handleResize(this.contentWidth);
        }
      } else {
        this.removeResize();
      }

      if (activeTabKey !== prevProps.activeTabKey && popupIsOpen) {
        // If the active tab has changed between updates due to updates outside of Tabs, the popup is closed.
        // eslint-disable-next-line react/no-did-update-set-state
        this.closePopup();
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      this.updateSize.cancel();
      this.removeResize();
    }
  }, {
    key: "getRollupTabWidth",
    value: function getRollupTabWidth() {
      return this.rollupTabRef.current.getBoundingClientRect().width;
    }
  }, {
    key: "closePopup",
    value: function closePopup(callback) {
      this.setState({
        popupIsOpen: false
      }, callback);
    }
  }, {
    key: "updateSize",
    value: function updateSize() {
      this.resetCalculations();
      this.forceUpdate();
    }
  }, {
    key: "resetCalculations",
    value: function resetCalculations() {
      this.hiddenStartIndex = -1;
      this.menuHidden = false;
      this.isCalculating = true;
    }
  }, {
    key: "initializeResize",
    value: function initializeResize() {
      var _this2 = this;

      if (!this.resizeListenerAdded) {
        this.resizeObserver = new _resizeObserverPolyfill.default(function (entries) {
          _this2.contentWidth = entries[0].contentRect.width;

          _this2.updateSize();
        });
        this.resizeObserver.observe(this.containerRef.current);
        this.resizeListenerAdded = true;
      }
    }
  }, {
    key: "removeResize",
    value: function removeResize() {
      if (this.resizeListenerAdded) {
        this.resizeObserver.disconnect(this.containerRef.current);
        this.resizeListenerAdded = false;
      }
    }
  }, {
    key: "handleResize",
    value: function handleResize(width) {
      // Calculate hide index
      var childrenCount = this.props.navigationItems.length;
      var moreWidth = width - this.getRollupTabWidth();
      var newHideIndex = childrenCount;
      var isMenuHidden = true;
      var calcMinWidth = 0;

      for (var i = 0; i < childrenCount; i += 1) {
        calcMinWidth += this.childRefs[i].current.getBoundingClientRect().width;

        if (calcMinWidth > moreWidth && !(i === childrenCount - 1 && calcMinWidth <= width)) {
          newHideIndex = i;
          isMenuHidden = false;
          break;
        }
      }

      if (this.hiddenStartIndex !== newHideIndex) {
        this.hiddenStartIndex = newHideIndex;
        this.menuHidden = isMenuHidden;
        this.forceUpdate();
      }
    }
  }, {
    key: "shouldPulse",
    value: function shouldPulse(navigationItems, notifications) {
      var shouldPulse = false;

      if (this.previousNotifications) {
        for (var i = 0; i < navigationItems.length; i += 1) {
          var item = navigationItems[i];
          var previousCount = this.previousNotifications[item.key];
          var newCount = notifications[item.key];

          if (newCount && (!previousCount || newCount > previousCount)) {
            shouldPulse = true;
            break;
          }
        }
      }

      return shouldPulse;
    }
  }, {
    key: "buildVisibleChildren",
    value: function buildVisibleChildren(visibleTabs, hasNotifications, onTabSelect, activeTabKey, notifications) {
      var _this3 = this;

      return visibleTabs.map(function (tab, index) {
        var tabProps = {
          text: tab.text,
          key: tab.key,
          onTabSelect: onTabSelect ? onTabSelect.bind(null, tab.key, tab.metaData) : null,
          isActive: tab.key === activeTabKey,
          notificationCount: hasNotifications ? notifications[tab.key] : 0,
          hasCount: hasNotifications
        };

        if (_this3.isCalculating) {
          var tabRef = _react.default.createRef();

          _this3.childRefs[index] = tabRef;
          tabProps.tabRef = tabRef;
        }

        return _react.default.createElement(_Tab.default, (0, _extends2.default)({}, tabProps, {
          render: _this3.props.navigationRenderFunction
        }));
      });
    }
  }, {
    key: "sliceTabs",
    value: function sliceTabs(navigationItems) {
      if (this.hiddenStartIndex >= 0) {
        return {
          visibleTabs: navigationItems.slice(0, this.hiddenStartIndex),
          hiddenTabs: navigationItems.slice(this.hiddenStartIndex)
        };
      }

      return {
        visibleTabs: navigationItems,
        hiddenTabs: []
      };
    }
  }, {
    key: "renderRollup",
    value: function renderRollup(hiddenTabs, hasNotifications, hasHiddenNotification) {
      var _this4 = this;

      var _this$props3 = this.props,
          activeTabKey = _this$props3.activeTabKey,
          notifications = _this$props3.notifications,
          intl = _this$props3.intl;
      var tabRollupIsSelected = hiddenTabs.some(function (tab) {
        return tab.key === activeTabKey;
      });
      return _react.default.createElement(_TabRollup.default, {
        hasCount: hasNotifications,
        isPulsed: hasHiddenNotification && !this.isCalculating && this.shouldPulse(hiddenTabs, notifications),
        onTabSelect: function onTabSelect() {
          _this4.setState({
            popupIsOpen: true
          });
        },
        tabRef: this.rollupTabRef,
        innerRef: this.rollupInnerRef,
        text: intl.formatMessage({
          id: 'Terra.applicationNavigation.tabs.rollupButtonTitle'
        }),
        isSelected: tabRollupIsSelected,
        hasChildNotifications: hasHiddenNotification,
        "data-application-tabs-more": true
      });
    }
  }, {
    key: "renderPopup",
    value: function renderPopup(hiddenTabs) {
      var _this5 = this;

      var _this$props4 = this.props,
          activeTabKey = _this$props4.activeTabKey,
          onTabSelect = _this$props4.onTabSelect,
          notifications = _this$props4.notifications,
          intl = _this$props4.intl;
      return _react.default.createElement(_terraPopup.default, {
        contentHeight: "auto",
        contentWidth: "320",
        onRequestClose: function onRequestClose() {
          _this5.closePopup();
        },
        targetRef: function targetRef() {
          return _this5.rollupInnerRef.current;
        },
        isOpen: true,
        isArrowDisplayed: true,
        isContentFocusDisabled: true
      }, _react.default.createElement(_PopupMenu.default, {
        title: intl.formatMessage({
          id: 'Terra.applicationNavigation.tabs.rollupMenuHeaderTitle'
        }),
        role: "list",
        menuItems: hiddenTabs.map(function (tab) {
          return {
            key: tab.key,
            text: tab.text,
            icon: tab.icon,
            notificationCount: notifications[tab.key],
            metaData: tab.metaData,
            isActive: tab.key === activeTabKey
          };
        }),
        onSelectMenuItem: function onSelectMenuItem(itemKey, itemMetaData) {
          _this5.closePopup(function () {
            if (onTabSelect) {
              onTabSelect(itemKey, itemMetaData);
            }
          });
        },
        showSelections: true
      }));
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props5 = this.props,
          navigationItems = _this$props5.navigationItems,
          activeTabKey = _this$props5.activeTabKey,
          onTabSelect = _this$props5.onTabSelect,
          notifications = _this$props5.notifications;

      if (!navigationItems || !navigationItems.length) {
        return _react.default.createElement(_Tab.default, {
          isPlaceholder: true,
          text: "W",
          tabKey: "",
          "aria-hidden": "true"
        });
      }

      var popupIsOpen = this.state.popupIsOpen;

      var _this$sliceTabs = this.sliceTabs(navigationItems),
          visibleTabs = _this$sliceTabs.visibleTabs,
          hiddenTabs = _this$sliceTabs.hiddenTabs;

      var hasVisibleNotification = visibleTabs.some(function (tab) {
        return !!notifications[tab.key];
      });
      var hasHiddenNotification = hiddenTabs.some(function (tab) {
        return !!notifications[tab.key];
      });
      var hasNotifications = hasVisibleNotification || hasHiddenNotification;
      return _react.default.createElement("nav", {
        className: cx('tabs-container', {
          'is-calculating': this.isCalculating
        }),
        ref: this.containerRef
      }, this.buildVisibleChildren(visibleTabs, hasNotifications, onTabSelect, activeTabKey, notifications), !this.menuHidden ? this.renderRollup(hiddenTabs, hasNotifications, hasHiddenNotification) : null, popupIsOpen ? this.renderPopup(hiddenTabs) : null);
    }
  }]);
  return Tabs;
}(_react.default.Component);

Tabs.propTypes = propTypes;
Tabs.defaultProps = defaultProps;

var _default = (0, _reactIntl.injectIntl)(Tabs);

exports.default = _default;
