// terra-modal-manager/lib/terra-dev-site/test/modal-manager/ModalManagerWithDisclosureAccessory.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _DisclosureComponent = _interopRequireDefault(require("./DisclosureComponent"));

var _ModalManager = _interopRequireDefault(require("../../../ModalManager"));

var _ModalManagerCommonTestModule = _interopRequireDefault(require("./ModalManagerCommon.test.module.scss"));

var cx = _bind.default.bind(_ModalManagerCommonTestModule.default);

var ModalManagerWithDisclosureAccessory = function ModalManagerWithDisclosureAccessory() {
  return _react.default.createElement("div", {
    role: "main",
    className: cx('content-wrapper')
  }, _react.default.createElement(_ModalManager.default, {
    disclosureAccessory: _react.default.createElement("div", {
      className: cx('disclosure-accessory')
    }, "Disclosure Accessory")
  }, _react.default.createElement(_DisclosureComponent.default, {
    identifier: "root-component",
    disclosureType: "modal",
    renderHeaderAdapter: true
  })));
};

var _default = ModalManagerWithDisclosureAccessory;
exports.default = _default;
