// terra-menu/lib/terra-dev-site/test/menu/menu/MediumMenu.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _Menu = _interopRequireDefault(require("../../../../Menu"));

var MediumMenu = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(MediumMenu, _React$Component);

  function MediumMenu(props) {
    var _this;

    (0, _classCallCheck2.default)(this, MediumMenu);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(MediumMenu).call(this, props));
    _this.handleButtonClick = _this.handleButtonClick.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleRequestClose = _this.handleRequestClose.bind((0, _assertThisInitialized2.default)(_this));
    _this.setButtonNode = _this.setButtonNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.getButtonNode = _this.getButtonNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.state = {
      open: false
    };
    return _this;
  }

  (0, _createClass2.default)(MediumMenu, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.forceUpdate();
    }
  }, {
    key: "setButtonNode",
    value: function setButtonNode(node) {
      this.buttonNode = node;
    }
  }, {
    key: "getButtonNode",
    value: function getButtonNode() {
      return this.buttonNode;
    }
  }, {
    key: "handleButtonClick",
    value: function handleButtonClick() {
      this.setState({
        open: true
      });
    }
  }, {
    key: "handleRequestClose",
    value: function handleRequestClose() {
      this.setState({
        open: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement("div", null, "This menu should have a medium height. And all items should be visible without scrolling."), _react.default.createElement(_Menu.default, {
        isOpen: this.state.open,
        targetRef: this.getButtonNode,
        onRequestClose: this.handleRequestClose
      }, _react.default.createElement(_Menu.default.Item, {
        text: "Default 1",
        key: "1",
        className: "TestFirstItem"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 2",
        key: "2"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 3",
        key: "3"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 4",
        key: "4"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 5",
        key: "5"
      }), _react.default.createElement(_Menu.default.ItemGroup, {
        key: "6"
      }, _react.default.createElement(_Menu.default.Item, {
        text: "Default 61",
        key: "61"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 62",
        key: "62"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 63",
        key: "63"
      })), _react.default.createElement(_Menu.default.Item, {
        text: "Default 7",
        key: "7"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 8",
        key: "8"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 9",
        key: "9"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 10",
        key: "10"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 11",
        key: "11"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 12",
        key: "12"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 13",
        key: "13"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 14",
        key: "14"
      }), _react.default.createElement(_Menu.default.ItemGroup, {
        key: "15"
      }, _react.default.createElement(_Menu.default.Item, {
        text: "Default 151",
        key: "151"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 152",
        key: "152"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 153",
        key: "153",
        className: "TestLastItem"
      }))), _react.default.createElement("button", {
        type: "button",
        id: "medium-menu-button",
        onClick: this.handleButtonClick,
        ref: this.setButtonNode
      }, "Default Menu"));
    }
  }]);
  return MediumMenu;
}(_react.default.Component);

var _default = MediumMenu;
exports.default = _default;
