// terra-menu/lib/terra-dev-site/test/menu/menu/SelectableAndUnselectableItemsMenu.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _Menu = _interopRequireDefault(require("../../../../Menu"));

var SelectableMenu = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(SelectableMenu, _React$Component);

  function SelectableMenu(props) {
    var _this;

    (0, _classCallCheck2.default)(this, SelectableMenu);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(SelectableMenu).call(this, props));
    _this.handleButtonClick = _this.handleButtonClick.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleRequestClose = _this.handleRequestClose.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleItemClick = _this.handleItemClick.bind((0, _assertThisInitialized2.default)(_this));
    _this.setButtonNode = _this.setButtonNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.getButtonNode = _this.getButtonNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.state = {
      open: false,
      isSelected: false
    };
    return _this;
  }

  (0, _createClass2.default)(SelectableMenu, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.forceUpdate();
    }
  }, {
    key: "setButtonNode",
    value: function setButtonNode(node) {
      this.buttonNode = node;
    }
  }, {
    key: "getButtonNode",
    value: function getButtonNode() {
      return this.buttonNode;
    }
  }, {
    key: "handleButtonClick",
    value: function handleButtonClick() {
      this.setState({
        open: true
      });
    }
  }, {
    key: "handleRequestClose",
    value: function handleRequestClose() {
      this.setState({
        open: false
      });
    }
  }, {
    key: "handleItemClick",
    value: function handleItemClick() {
      this.setState(function (prevState) {
        return {
          isSelected: !prevState.isSelected
        };
      });
      this.handleRequestClose();
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement("div", null, "This menu contains a selectable item and an unselectable item. All items in the menu should have the same spacing on the left to allow for a checkmark."), _react.default.createElement(_Menu.default, {
        isOpen: this.state.open,
        targetRef: this.getButtonNode,
        onRequestClose: this.handleRequestClose
      }, _react.default.createElement(_Menu.default.Item, {
        text: "Default 1",
        key: "1",
        className: "TestNonSelectableItem"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 2",
        key: "2",
        isSelectable: true,
        className: "TestSelectableItem",
        isSelected: this.state.isSelected,
        onClick: this.handleItemClick
      })), _react.default.createElement("button", {
        type: "button",
        id: "default-button",
        onClick: this.handleButtonClick,
        ref: this.setButtonNode
      }, "Default Menu"));
    }
  }]);
  return SelectableMenu;
}(_react.default.Component);

var _default = SelectableMenu;
exports.default = _default;
