// terra-menu/lib/terra-dev-site/test/menu/menu/SubMenu.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _Menu = _interopRequireDefault(require("../../../../Menu"));

var SubMenu = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(SubMenu, _React$Component);

  function SubMenu(props) {
    var _this;

    (0, _classCallCheck2.default)(this, SubMenu);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(SubMenu).call(this, props));
    _this.handleButtonClick = _this.handleButtonClick.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleRequestClose = _this.handleRequestClose.bind((0, _assertThisInitialized2.default)(_this));
    _this.setButtonNode = _this.setButtonNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.getButtonNode = _this.getButtonNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.state = {
      open: false
    };
    return _this;
  }

  (0, _createClass2.default)(SubMenu, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.forceUpdate();
    }
  }, {
    key: "setButtonNode",
    value: function setButtonNode(node) {
      this.buttonNode = node;
    }
  }, {
    key: "getButtonNode",
    value: function getButtonNode() {
      return this.buttonNode;
    }
  }, {
    key: "handleButtonClick",
    value: function handleButtonClick() {
      this.setState({
        open: true
      });
    }
  }, {
    key: "handleRequestClose",
    value: function handleRequestClose() {
      this.setState({
        open: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement("div", null, "This menu has a nested submenu. When the Nested Menu item is clicked as submenu should replace the initial menu. There should be a header with a back button and a title of Nested Menu."), _react.default.createElement(_Menu.default, {
        isOpen: this.state.open,
        targetRef: this.getButtonNode,
        onRequestClose: this.handleRequestClose,
        headerTitle: "Sample header"
      }, _react.default.createElement(_Menu.default.Item, {
        text: "Default 1",
        key: "1",
        className: "TestInitialMenuContent"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Nested Menu",
        key: "2",
        className: "TestNestedMenu",
        subMenuItems: [_react.default.createElement(_Menu.default.Item, {
          text: "Default 2.1",
          key: "2.1",
          className: "TestNestedMenuContent"
        })]
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 3",
        key: "3"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 4",
        key: "4"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Default 5",
        key: "5"
      })), _react.default.createElement("button", {
        type: "button",
        id: "sub-menu-button",
        onClick: this.handleButtonClick,
        ref: this.setButtonNode
      }, "Default Menu"));
    }
  }]);
  return SubMenu;
}(_react.default.Component);

var _default = SubMenu;
exports.default = _default;
