// terra-menu/lib/terra-dev-site/doc/example/MenuBounded.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _BasicMenu = _interopRequireDefault(require("./BasicMenu"));

var _MenuBoundedModule = _interopRequireDefault(require("./MenuBounded.module.scss"));

var cx = _bind.default.bind(_MenuBoundedModule.default);

var MenuBounded = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(MenuBounded, _React$Component);

  function MenuBounded(props) {
    var _this;

    (0, _classCallCheck2.default)(this, MenuBounded);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(MenuBounded).call(this, props));
    _this.setParentNode = _this.setParentNode.bind((0, _assertThisInitialized2.default)(_this));
    _this.getParentNode = _this.getParentNode.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(MenuBounded, [{
    key: "setParentNode",
    value: function setParentNode(node) {
      this.parentNode = node;
    }
  }, {
    key: "getParentNode",
    value: function getParentNode() {
      return this.parentNode;
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", {
        className: cx('content-wrapper'),
        ref: this.setParentNode
      }, _react.default.createElement(_BasicMenu.default, {
        boundingRef: this.getParentNode
      }));
    }
  }]);
  return MenuBounded;
}(_react.default.Component);

var _default = MenuBounded;
exports.default = _default;
