// terra-menu/lib/terra-dev-site/doc/example/MenuWidths.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _BasicMenu = _interopRequireDefault(require("./BasicMenu"));

var MenuWidths = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(MenuWidths, _React$Component);

  function MenuWidths(props) {
    var _this;

    (0, _classCallCheck2.default)(this, MenuWidths);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(MenuWidths).call(this, props));
    _this.handleSelectChange = _this.handleSelectChange.bind((0, _assertThisInitialized2.default)(_this));
    _this.state = {
      menuWidth: undefined
    };
    return _this;
  }

  (0, _createClass2.default)(MenuWidths, [{
    key: "handleSelectChange",
    value: function handleSelectChange(event) {
      this.setState({
        menuWidth: event.target.value
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement("label", {
        htmlFor: "menuWidth"
      }, "Select menu width "), _react.default.createElement("select", {
        id: "menuWidth",
        name: "menuWidth",
        value: this.state.menuWidth,
        onChange: this.handleSelectChange
      }, _react.default.createElement("option", {
        value: undefined
      }, "Default"), _react.default.createElement("option", {
        key: "160",
        value: "160"
      }, "160"), _react.default.createElement("option", {
        key: "240",
        value: "240"
      }, "240"), _react.default.createElement("option", {
        key: "320",
        value: "320"
      }, "320"), _react.default.createElement("option", {
        key: "640",
        value: "640"
      }, "640"), _react.default.createElement("option", {
        key: "960",
        value: "960"
      }, "960"), _react.default.createElement("option", {
        key: "1280",
        value: "1280"
      }, "1280"), _react.default.createElement("option", {
        key: "1760",
        value: "1760"
      }, "1760"), _react.default.createElement("option", {
        key: "auto",
        value: "auto"
      }, "Auto")), _react.default.createElement("br", null), _react.default.createElement("br", null), _react.default.createElement(_BasicMenu.default, {
        contentWidth: this.state.menuWidth
      }));
    }
  }]);
  return MenuWidths;
}(_react.default.Component);

var _default = MenuWidths;
exports.default = _default;
