// terra-markdown/lib/Markdown.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _marked = _interopRequireDefault(require("marked"));

var _prismjs = _interopRequireDefault(require("prismjs"));

require("prismjs/components/prism-diff");

require("prismjs/components/prism-jsx");

require("prismjs/components/prism-scss");

var _Markdown = _interopRequireDefault(require("./Markdown.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var cx = _bind.default.bind(_Markdown.default); // Create a list of loaded languages, remove functions that aren't actaully languages.


var supportedLanguages = Object.keys(_prismjs.default.languages).filter(function (lang) {
  return !['extend', 'insertBefore', 'DFS'].includes(lang);
}); // If the supported language is requested, highlight it.

var highlight = function highlight(code, lang) {
  if (supportedLanguages.includes(lang)) {
    return _prismjs.default.highlight(code, _prismjs.default.languages[lang], lang);
  }

  return null;
};

_marked.default.setOptions({
  headerIds: false,
  highlight: highlight,
  langPrefix: 'codeblock language-'
});

var renderer = new _marked.default.Renderer(); // Override how heading renders to add the anchor.

renderer.heading = function (text, level) {
  var escapedText = text.toLowerCase().replace(/[^\w]+/g, '-');
  return "<h".concat(level, ">") + "<a name=\"".concat(escapedText, "\" class=\"anchor\" aria-hidden=\"true\" href=\"#").concat(escapedText, "\">") + '<svg data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" class="is-bidi octicon octicon-link"><path d="M10.5 26.3a16 16 0 0 1 .9-5.3L8 17.6a6.73 6.73 0 0 1-2-4.8A6.41 6.41 0 0 1 8 8a6.76 6.76 0 0 1 9.6 0l9 9a6.88 6.88 0 0 1 0 9.6 6.27 6.27 0 0 1-3.4 1.8 2.65 2.65 0 0 0 .5.6l4.1 4.1a11.36 11.36 0 0 0 3.1-2.3 12.8 12.8 0 0 0 0-18.1l-9-9a12.63 12.63 0 0 0-9-3.7 12.63 12.63 0 0 0-9 3.7A12.1 12.1 0 0 0 0 12.8a12.63 12.63 0 0 0 3.7 9l7 7a21.05 21.05 0 0 1-.2-2.5zm27-4.6a16 16 0 0 1-.9 5.3l3.4 3.4a6.79 6.79 0 1 1-9.6 9.6l-9-9a6.88 6.88 0 0 1 0-9.6 6.27 6.27 0 0 1 3.4-1.8 2.65 2.65 0 0 0-.5-.6l-4-4a11.36 11.36 0 0 0-3.1 2.3 12.8 12.8 0 0 0 0 18.1l9 9a12.73 12.73 0 0 0 18-18l-7-7a8 8 0 0 1 .3 2.3z"/></svg>' + '</a>' + "".concat(text) + "</h".concat(level, ">");
};

var propTypes = {
  /**
   * Markdown content to parse.
   */
  src: _propTypes.default.string,

  /**
   * A prefix url for any relative link.
   */
  baseUrl: _propTypes.default.string,

  /**
   * Enable anchors for heading elements.
   */
  hasHeadingAnchors: _propTypes.default.bool
};
var defaultProps = {
  src: ''
};
/* eslint react/no-danger:0 */

var Markdown = function Markdown(props) {
  return _react.default.createElement("div", {
    dir: "ltr",
    className: cx(['markdown']),
    dangerouslySetInnerHTML: {
      __html: (0, _marked.default)(props.src, _objectSpread({}, props.baseUrl && {
        baseUrl: props.baseUrl
      }, {}, props.hasHeadingAnchors && {
        renderer: renderer
      }))
    }
  });
};

Markdown.propTypes = propTypes;
Markdown.defaultProps = defaultProps;
var _default = Markdown;
exports.default = _default;
