// terra-application/lib/terra-dev-site/test/slide-panel-manager/NavigationPrompt.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _DisclosureComponent = _interopRequireDefault(require("./DisclosureComponent"));

var _slidePanelManager = _interopRequireDefault(require("../../../slide-panel-manager"));

var _SlidePanelManagerDefaultTestModule = _interopRequireDefault(require("./SlidePanelManagerDefault.test.module.scss"));

var cx = _bind.default.bind(_SlidePanelManagerDefaultTestModule.default);

var SlidePanelManagerDefault = function SlidePanelManagerDefault() {
  return _react.default.createElement("div", {
    role: "main",
    className: cx('content-wrapper')
  }, _react.default.createElement(_slidePanelManager.default, null, _react.default.createElement(_DisclosureComponent.default, {
    identifier: "root-component",
    disclosureType: "panel",
    renderHeaderAdapter: true
  })));
};

var _default = SlidePanelManagerDefault;
exports.default = _default;
