// terra-menu/lib/terra-dev-site/test/menu/menu-item-group/MenuItemGroupDefault.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Menu = _interopRequireDefault(require("../../../../Menu"));

var childContextTypes = {
  isSelectableMenu: _propTypes.default.bool
};

var DefaultMenuItemGroup = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(DefaultMenuItemGroup, _React$Component);

  function DefaultMenuItemGroup(props) {
    var _this;

    (0, _classCallCheck2.default)(this, DefaultMenuItemGroup);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(DefaultMenuItemGroup).call(this, props));
    _this.state = {
      selectedIndex: null
    };
    _this.handleSelection = _this.handleSelection.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(DefaultMenuItemGroup, [{
    key: "getChildContext",
    value: function getChildContext() {
      return {
        isSelectableMenu: true
      };
    }
  }, {
    key: "handleSelection",
    value: function handleSelection(event, selectedIndex) {
      this.setState({
        selectedIndex: selectedIndex
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement("div", {
        id: "selected-index"
      }, _react.default.createElement("h3", null, "Selected Button:", this.state.selectedIndex)), _react.default.createElement(_Menu.default.ItemGroup, {
        className: "TestGroup",
        onChange: this.handleSelection
      }, _react.default.createElement(_Menu.default.Item, {
        text: "Group Item 1",
        key: "1",
        className: "TestGroupItem1"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Group Item 2",
        key: "2",
        className: "TestGroupItem2"
      }), _react.default.createElement(_Menu.default.Item, {
        text: "Group Item 3",
        key: "3",
        className: "TestGroupItem3"
      })));
    }
  }]);
  return DefaultMenuItemGroup;
}(_react.default.Component);

DefaultMenuItemGroup.childContextTypes = childContextTypes;
var _default = DefaultMenuItemGroup;
exports.default = _default;
