// terra-overlay/lib/terra-dev-site/test/overlay/Overlay/ContainerScrollableOverlay.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _OverlayContainer = _interopRequireDefault(require("../../../../OverlayContainer"));

var _Overlay = _interopRequireDefault(require("../../../../Overlay"));

var _OverlayTestCommonModule = _interopRequireDefault(require("./OverlayTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_OverlayTestCommonModule.default);

var OverlayExample = function OverlayExample() {
  return _react.default.createElement(_OverlayContainer.default, {
    className: cx('overlay-container2'),
    id: "overlay-container"
  }, _react.default.createElement(_Overlay.default, {
    isOpen: true,
    isRelativeToContainer: true,
    isScrollable: true
  }, _react.default.createElement("p", null, "I"), _react.default.createElement("p", null, "am"), _react.default.createElement("p", null, "some"), _react.default.createElement("p", null, "very"), _react.default.createElement("p", null, "tall"), _react.default.createElement("p", null, "content"), _react.default.createElement("p", null, "in"), _react.default.createElement("p", null, "the"), _react.default.createElement("p", null, "overlay"), _react.default.createElement("p", null, "to"), _react.default.createElement("p", null, "prove"), _react.default.createElement("p", null, "that"), _react.default.createElement("p", null, "isScrollable"), _react.default.createElement("p", null, "works.")));
};

var _default = OverlayExample;
exports.default = _default;
