// terra-overlay/lib/terra-dev-site/test/overlay/Overlay/LightOverlay.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _OverlayContainer = _interopRequireDefault(require("../../../../OverlayContainer"));

var _Overlay = _interopRequireDefault(require("../../../../Overlay"));

var _OverlayTestCommonModule = _interopRequireDefault(require("./OverlayTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_OverlayTestCommonModule.default);

var OverlayComponent = function OverlayComponent() {
  return _react.default.createElement(_Overlay.default, {
    isOpen: true,
    isRelativeToContainer: true,
    backgroundStyle: _Overlay.default.Opts.BackgroundStyles.LIGHT,
    id: "light-overlay"
  }, _react.default.createElement("h1", null, "Light Overlay"));
};

var OverlayExample = function OverlayExample() {
  return _react.default.createElement(_OverlayContainer.default, {
    className: cx('overlay-container1'),
    id: "terra-OverlayContainer",
    overlay: _react.default.createElement(OverlayComponent, null)
  });
};

var _default = OverlayExample;
exports.default = _default;
