// terra-application/lib/navigation-prompt/getUnsavedChangesPromptOptions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

/**
 * Generates an Object containing properties used to define messages for NavigationPromptCheckpoint-based user prompting.
 * @param {Object} intl - The react-intl provided intl object used to look up translated string definitions.
 */
var _default = function _default(intl) {
  return function (prompts) {
    var startMessage;
    var content;
    var endMessage;

    if (prompts.length === 1) {
      startMessage = intl.formatMessage({
        id: 'terraApplication.unsavedChangesPrompt.singlePromptMessage'
      }, {
        promptDescription: prompts[0].description
      });
    } else {
      startMessage = intl.formatMessage({
        id: 'terraApplication.unsavedChangesPrompt.multiplePromptMessageIntro'
      }); // We do not currently have access to the prompts internal identifiers.

      /* eslint-disable react/no-array-index-key */

      content = _react.default.createElement("ul", null, prompts.map(function (prompt, index) {
        return _react.default.createElement("li", {
          key: index
        }, prompt.description);
      }));
      /* eslint-enable react/no-array-index-key */

      endMessage = intl.formatMessage({
        id: 'terraApplication.unsavedChangesPrompt.multiplePromptMessageOutro'
      });
    }

    return {
      title: intl.formatMessage({
        id: 'terraApplication.unsavedChangesPrompt.title'
      }),
      startMessage: startMessage,
      content: content,
      endMessage: endMessage,
      buttonOrder: 'rejectFirst',
      emphasizedAction: 'reject',
      acceptButtonText: intl.formatMessage({
        id: 'terraApplication.unsavedChangesPrompt.acceptButtonText'
      }),
      rejectButtonText: intl.formatMessage({
        id: 'terraApplication.unsavedChangesPrompt.rejectButtonText'
      })
    };
  };
};

exports.default = _default;
