// terra-application/lib/navigation-prompt/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "getUnsavedChangesPromptOptions", {
  enumerable: true,
  get: function get() {
    return _getUnsavedChangesPromptOptions.default;
  }
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _terraNavigationPrompt.default;
  }
});
Object.defineProperty(exports, "NavigationPromptCheckpoint", {
  enumerable: true,
  get: function get() {
    return _terraNavigationPrompt.NavigationPromptCheckpoint;
  }
});
Object.defineProperty(exports, "PromptRegistrationContext", {
  enumerable: true,
  get: function get() {
    return _terraNavigationPrompt.PromptRegistrationContext;
  }
});
exports.navigationPromptResolutionOptionsShape = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _getUnsavedChangesPromptOptions = _interopRequireDefault(require("./getUnsavedChangesPromptOptions"));

var _terraNavigationPrompt = _interopRequireWildcard(require("terra-navigation-prompt"));

var navigationPromptResolutionOptionsShape = _propTypes.default.oneOfType([_propTypes.default.shape({
  /**
   * The title string for the NotificationDialog.
   */
  title: _propTypes.default.string,

  /**
   * The startMessage string for the NotificationDialog.
   */
  startMessage: _propTypes.default.string,

  /**
   * The content element for the NotificationDialog.
   */
  content: _propTypes.default.node,

  /**
   * The endMessage string for the NotificationDialog.
   */
  endMessage: _propTypes.default.string,

  /**
   * The string for the NotificationDialog's accept button text.
   */
  acceptButtonText: _propTypes.default.string,

  /**
   * The string for the NotificationDialog's reject button text.
   */
  rejectButtonText: _propTypes.default.string,

  /**
   * The string specifying the emphasized action.
   */
  emphasizedAction: _propTypes.default.oneOf(['accept', 'reject', 'none']),

  /**
   * The string specifying the button order.
   */
  buttonOrder: _propTypes.default.oneOf(['acceptFirst', 'rejectFirst'])
}), _propTypes.default.func]);

exports.navigationPromptResolutionOptionsShape = navigationPromptResolutionOptionsShape;
