// terra-doc-template/lib/terra-dev-site/test/doc-template/DefaultDocTemplate.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _DocTemplate = _interopRequireDefault(require("../../../DocTemplate"));

var _TestReadme = _interopRequireDefault(require("./common/TestReadme.md"));

var _package = require("../../../../package.json");

var _TestComponentExample = _interopRequireDefault(require("./common/TestComponentExample"));

var _CustomPropsTable = _interopRequireDefault(require("./common/CustomPropsTable"));

var _TestComponentExample2 = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/test/doc-template/common/TestComponentExample"));

var _TestComponent = _interopRequireDefault(require("!raw-loader!../../../../src/terra-dev-site/test/doc-template/common/TestComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable import/no-webpack-loader-syntax, import/first, import/no-duplicates */

/* eslint-enisable import/no-webpack-loader-syntax, import/first */
var Index = function Index() {
  var propsTables = [{
    componentSrc: _TestComponent.default,
    componentName: 'Test Component'
  }, {
    componentSrc: _TestComponent.default,
    componentName: 'Test Component (Again)'
  }];

  var exampleElement = _react.default.createElement(_TestComponentExample.default, null);

  var examples = [{
    title: 'Test Example 1',
    description: 'Describing the test',
    example: exampleElement,
    source: _TestComponentExample2.default
  }, {
    title: 'Test Example 2',
    description: 'Describing the test mk. 2',
    example: exampleElement,
    source: _TestComponentExample2.default
  }];
  return _react.default.createElement("div", {
    id: "DocTemplateContainer"
  }, _react.default.createElement(_DocTemplate.default, {
    packageName: _package.name,
    packageVersion: _package.version,
    readme: _TestReadme.default,
    srcPath: "https://github.com/cerner/terra-core/tree/master/packages/terra-doc-template",
    propsTables: propsTables,
    examples: examples
  }, _react.default.createElement(_CustomPropsTable.default, null)));
};

var _default = Index;
exports.default = _default;
