// terra-form-select/lib/shared/_FrameUtil.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _variants = _interopRequireDefault(require("./_variants"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

var FrameUtil = /*#__PURE__*/function () {
  function FrameUtil() {
    _classCallCheck(this, FrameUtil);
  }

  _createClass(FrameUtil, null, [{
    key: "dropdownStyle",

    /**
     * Returns the dropdown style.
     * @param {Object} props - The component props.
     * @param {Object} state - The component state.
     * @return {Object} - The dropdown style.
     */
    value: function dropdownStyle() {
      var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
      var state = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
      var style = props.style;
      var maxHeight = state.maxHeight,
          width = state.width,
          top = state.top;
      return _objectSpread({}, style, {
        maxHeight: maxHeight,
        width: width,
        top: top
      });
    }
    /**
     * Determines the dropdown position.
     * @param {Object} props - The component props.
     * @param {ReactNode} target - The select wrapper.
     * @param {ReactNode} dropdown - The dropdown.
     * @param {number} maxHeight - The maxHeight of the dropdown dropdown.
     * @param {boolean} useSemanticDropdown - If the dropdown should be rendered semantically instead of in a portal
     * @return {Object} - The calculated dropdown attributes.
     */

  }, {
    key: "dropdownPosition",
    value: function dropdownPosition(props, target, dropdown, maxHeight) {
      var useSemanticDropdown = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : false;
      var style = _objectSpread({}, props).style || {};

      var _dropdown$getBounding = dropdown.getBoundingClientRect(),
          height = _dropdown$getBounding.height;

      var _target$getBoundingCl = target.getBoundingClientRect(),
          bottom = _target$getBoundingCl.bottom,
          width = _target$getBoundingCl.width,
          top = _target$getBoundingCl.top;

      var spaceBelow = window.innerHeight - bottom;
      var maximumHeight = parseInt(style.maxHeight || maxHeight, 10) || Infinity;
      var canFitBelow = maximumHeight < spaceBelow || height < spaceBelow || spaceBelow > top;
      var availableSpace = canFitBelow ? spaceBelow : top;
      var availableMaxHeight = Math.min(maximumHeight, availableSpace - 10);
      var isAbove = !canFitBelow;
      var semanticTopWhenAbove = useSemanticDropdown && isAbove ? -1 * Math.min(availableMaxHeight, height) : undefined;
      return {
        width: width,
        maxHeight: availableMaxHeight,
        isAbove: isAbove,
        isPositioned: true,
        top: semanticTopWhenAbove
      };
    }
    /**
     * Determines whether the variant allows multiple selections.
     * @param {Object} variant - The component props.
     * @return {boolean} - True if the variant allows multiple selections.
     */

  }, {
    key: "allowsMultiSelections",
    value: function allowsMultiSelections(props) {
      return props.variant === _variants.default.MULTIPLE || props.variant === _variants.default.TAG;
    }
    /**
     * Determines if the query is included in the component value.
     * @param {Object} props - The component props.
     * @param {string} query - The query value.
     * @return {boolean} - True if the query is contained within the component value.
     */

  }, {
    key: "includes",
    value: function includes(props, query) {
      if (query.trim().length === 0) {
        return false;
      }

      return (props.value || []).indexOf(query) > -1;
    }
    /**
     * Determines whether the search input should be hidden.
     * @param {Object} props - The component props;
     * @param {Object} state - The component state;
     * @return {Boolean} - True if the search input should be hidden.
     */

  }, {
    key: "shouldHideSearch",
    value: function shouldHideSearch(props, state) {
      if (FrameUtil.allowsMultiSelections(props)) {
        return !state.isFocused && props.value && props.value.length > 0;
      }

      return false;
    }
    /**
     * Determines whether the dropdown should be positioned.
     * @param {Object} previousState - The previous component state.
     * @param {Object} currentState - The current component state.
     * @param {ReactNode} dropdown - The component dropdown.
     * @return {boolean} - True if the dropdown should be positioned.
     */

  }, {
    key: "shouldPositionDropdown",
    value: function shouldPositionDropdown(previousState, currentState, dropdown) {
      if (!currentState.isOpen) {
        return false;
      }

      var _dropdown$getBounding2 = dropdown.getBoundingClientRect(),
          bottom = _dropdown$getBounding2.bottom;

      return previousState.isOpen === false || bottom > window.innerHeight;
    }
    /**
     * Determines whether the option should be added on blur.
     * @param {Object} props - The component props.
     * @param {Object} state - The component state;
     * @return {boolean} - True if the option should be added.
     */

  }, {
    key: "shouldAddOptionOnBlur",
    value: function shouldAddOptionOnBlur(props, state) {
      var onSelect = props.onSelect,
          value = props.value,
          variant = props.variant;
      var hasSearchChanged = state.hasSearchChanged,
          searchValue = state.searchValue;

      if ((variant === _variants.default.TAG || variant === _variants.default.COMBOBOX) && (variant !== _variants.default.TAG || searchValue.trim().length > 0) && hasSearchChanged && onSelect && (variant !== _variants.default.TAG || !FrameUtil.includes(props, searchValue)) && (variant !== _variants.default.COMBOBOX || value !== searchValue)) {
        return true;
      }

      return false;
    }
    /**
     * Determines the variant appropriate container tab index.
     * @param {Object} props - The component props.
     * @return {string} - A tab index.
     */

  }, {
    key: "tabIndex",
    value: function tabIndex(props) {
      if (props.variant === _variants.default.DEFAULT && !props.disabled) {
        return '0';
      }

      return '-1';
    }
  }]);

  return FrameUtil;
}();

var _default = FrameUtil;
exports.default = _default;
