// terra-application/lib/terra-dev-site/doc/application/reference.1/components/example/ApplicationLoadingOverlayExample.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _ApplicationLoadingOverlay = _interopRequireDefault(require("terra-application/lib/application-loading-overlay/ApplicationLoadingOverlay"));

var _ApplicationLoadingOverlayProvider = _interopRequireDefault(require("terra-application/lib/application-loading-overlay/ApplicationLoadingOverlayProvider"));

var _ApplicationErrorBoundaryExampleModule = _interopRequireDefault(require("./ApplicationErrorBoundaryExample.module.scss"));

var cx = _bind.default.bind(_ApplicationErrorBoundaryExampleModule.default);

var ApplicationLoadingOverlayExample = function ApplicationLoadingOverlayExample() {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      showClearOverlay = _useState2[0],
      setShowClearOverlay = _useState2[1];

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
      showLightOverlay = _useState4[0],
      setShowLightOverlay = _useState4[1];

  var _useState5 = (0, _react.useState)(false),
      _useState6 = (0, _slicedToArray2.default)(_useState5, 2),
      showDarkOverlay = _useState6[0],
      setShowDarkOverlay = _useState6[1];

  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("button", {
    type: "button",
    onClick: function onClick() {
      setShowClearOverlay(true);
      setTimeout(function () {
        setShowClearOverlay(false);
      }, 3000);
    }
  }, "Show Clear Loading Overlay"), _react.default.createElement("button", {
    type: "button",
    onClick: function onClick() {
      setShowLightOverlay(true);
      setTimeout(function () {
        setShowLightOverlay(false);
      }, 3000);
    }
  }, "Show Light Loading Overlay"), _react.default.createElement("button", {
    type: "button",
    onClick: function onClick() {
      setShowDarkOverlay(true);
      setTimeout(function () {
        setShowDarkOverlay(false);
      }, 3000);
    }
  }, "Show Dark Loading Overlay"), _react.default.createElement("div", {
    className: cx('example')
  }, _react.default.createElement(_ApplicationLoadingOverlayProvider.default, null, _react.default.createElement("p", null, "For this example, the loading overlays will be active for 3 seconds."), _react.default.createElement(_ApplicationLoadingOverlay.default, {
    isOpen: showClearOverlay
  }), _react.default.createElement(_ApplicationLoadingOverlay.default, {
    isOpen: showLightOverlay,
    backgroundStyle: "light"
  }), _react.default.createElement(_ApplicationLoadingOverlay.default, {
    isOpen: showDarkOverlay,
    backgroundStyle: "dark"
  }))));
};

var _default = ApplicationLoadingOverlayExample;
exports.default = _default;
