// terra-application/lib/terra-dev-site/doc/application/reference.1/components/example/ApplicationErrorBoundaryExample.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _ApplicationErrorBoundary = _interopRequireDefault(require("terra-application/lib/application-error-boundary/ApplicationErrorBoundary"));

var _ApplicationErrorBoundaryExampleModule = _interopRequireDefault(require("./ApplicationErrorBoundaryExample.module.scss"));

var cx = _bind.default.bind(_ApplicationErrorBoundaryExampleModule.default);

var ErrorThrowingComponent = function ErrorThrowingComponent() {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      throwError = _useState2[0],
      setThrowError = _useState2[1];

  if (throwError) {
    throw new Error('whoops');
  }

  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("p", null, "Press the button below to trigger an error."), _react.default.createElement("button", {
    type: "button",
    onClick: function onClick() {
      setThrowError(true);
    }
  }, "Trigger Error"));
};

var ApplicationErrorBoundaryExampleWrapper = function ApplicationErrorBoundaryExampleWrapper(_ref) {
  var children = _ref.children;

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
      refresh = _useState4[0],
      setRefresh = _useState4[1];

  return _react.default.createElement("div", {
    className: cx('example')
  }, _react.default.createElement("button", {
    type: "button",
    onClick: function onClick() {
      setRefresh(!refresh);
    }
  }, "Refresh Component"), _react.default.createElement(_ApplicationErrorBoundary.default, null, children));
};

ApplicationErrorBoundaryExampleWrapper.propTypes = {
  children: _propTypes.default.node
};

var ApplicationErrorBoundaryExample = function ApplicationErrorBoundaryExample() {
  return _react.default.createElement(ApplicationErrorBoundaryExampleWrapper, null, _react.default.createElement(ErrorThrowingComponent, null));
};

var _default = ApplicationErrorBoundaryExample;
exports.default = _default;
