// terra-hookshot/lib/terra-dev-site/test/hookshot/DefaultHookshot.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Hookshot = _interopRequireDefault(require("../../../Hookshot"));

var _HookshotTestDocCommonModule = _interopRequireDefault(require("../common/HookshotTestDocCommon.module.scss"));

var cx = _bind.default.bind(_HookshotTestDocCommonModule.default);

var HookshotStandard =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(HookshotStandard, _React$Component);

  function HookshotStandard(props) {
    var _this;

    (0, _classCallCheck2.default)(this, HookshotStandard);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(HookshotStandard).call(this, props));
    _this.handleButtonClick = _this.handleButtonClick.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleRequestClose = _this.handleRequestClose.bind((0, _assertThisInitialized2.default)(_this));
    _this.state = {
      open: false
    };
    return _this;
  }

  (0, _createClass2.default)(HookshotStandard, [{
    key: "handleButtonClick",
    value: function handleButtonClick() {
      this.setState({
        open: true
      });
    }
  }, {
    key: "handleRequestClose",
    value: function handleRequestClose() {
      this.setState({
        open: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      var hookshotContent = _react.default.createElement(_Hookshot.default.Content, {
        onEsc: this.handleRequestClose,
        onOutsideClick: this.handleRequestClose,
        onResize: this.handleRequestClose,
        id: "testDefaultContent"
      }, _react.default.createElement("div", {
        className: cx('default-hookshot-wrapper')
      }, "Hookshot"));

      return _react.default.createElement("div", {
        id: "default-bounds",
        className: cx('content-wrapper')
      }, _react.default.createElement(_Hookshot.default, {
        contentAttachment: {
          vertical: 'bottom',
          horizontal: 'center'
        },
        isEnabled: true,
        isOpen: this.state.open,
        targetRef: function targetRef() {
          return document.getElementById('hookshot-standard-button');
        }
      }, hookshotContent), _react.default.createElement("button", {
        type: "button",
        id: "hookshot-standard-button",
        onClick: this.handleButtonClick
      }, "Default Hookshot"));
    }
  }]);
  return HookshotStandard;
}(_react.default.Component);

var _default = HookshotStandard;
exports.default = _default;
