// terra-hookshot/lib/terra-dev-site/test/hookshot/HookshotCoordinates.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Hookshot = _interopRequireDefault(require("../../../Hookshot"));

var _HookshotContentTestTemplate = _interopRequireDefault(require("../common/HookshotContentTestTemplate"));

var _HookshotTestDocCommonModule = _interopRequireDefault(require("../common/HookshotTestDocCommon.module.scss"));

var cx = _bind.default.bind(_HookshotTestDocCommonModule.default);

var HookshotTemplate =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(HookshotTemplate, _React$Component);

  function HookshotTemplate(props) {
    var _this;

    (0, _classCallCheck2.default)(this, HookshotTemplate);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(HookshotTemplate).call(this, props));
    _this.triggerHookshot = _this.triggerHookshot.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleRequestClose = _this.handleRequestClose.bind((0, _assertThisInitialized2.default)(_this));
    _this.state = {
      open: false
    };
    return _this;
  }

  (0, _createClass2.default)(HookshotTemplate, [{
    key: "triggerHookshot",
    value: function triggerHookshot() {
      this.setState({
        open: true
      });
    }
  }, {
    key: "handleRequestClose",
    value: function handleRequestClose() {
      this.setState({
        open: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", {
        id: "coords-test",
        className: cx('hookshot-wrapper')
      }, _react.default.createElement(_Hookshot.default, {
        contentAttachment: {
          vertical: 'top',
          horizontal: 'start'
        },
        targetCoordinates: {
          x: 50,
          y: 132
        },
        isEnabled: true,
        isOpen: this.state.open
      }, _react.default.createElement(_HookshotContentTestTemplate.default, {
        id: "test-coords-content",
        onEsc: this.handleRequestClose,
        onOutsideClick: this.handleRequestClose
      })), _react.default.createElement("button", {
        type: "button",
        id: "coords-button",
        onClick: this.triggerHookshot
      }, "Trigger Hookshot"));
    }
  }]);
  return HookshotTemplate;
}(_react.default.Component);

var _default = HookshotTemplate;
exports.default = _default;
