// terra-hookshot/lib/terra-dev-site/test/hookshot/HookshotCloseBehaviors.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _HookshotTestTemplate = _interopRequireDefault(require("../common/HookshotTestTemplate"));

var NoCloseBehavior = function NoCloseBehavior() {
  return _react.default.createElement(_HookshotTestTemplate.default, {
    id: "NoCloseBehavior",
    type: "primary",
    hookshotContentProps: {
      closeOnEsc: false,
      closeOnOutsideClick: false,
      closeOnResize: false
    }
  });
};

var EscBehavior = function EscBehavior() {
  return _react.default.createElement(_HookshotTestTemplate.default, {
    id: "EscBehavior",
    type: "primary",
    hookshotContentProps: {
      closeOnEsc: true,
      closeOnOutsideClick: false,
      closeOnResize: false
    }
  });
};

var OutsideClickBehavior = function OutsideClickBehavior() {
  return _react.default.createElement(_HookshotTestTemplate.default, {
    id: "OutsideClickBehavior",
    type: "primary",
    hookshotContentProps: {
      closeOnEsc: false,
      closeOnOutsideClick: true,
      closeOnResize: false
    }
  });
};

var OutsideClickBehaviorWithSvgs = function OutsideClickBehaviorWithSvgs() {
  return _react.default.createElement(_HookshotTestTemplate.default, {
    id: "OutsideClickBehaviorWithSvgs",
    type: "primary",
    includeSvgs: true,
    hookshotContentProps: {
      closeOnEsc: false,
      closeOnOutsideClick: true,
      closeOnResize: false
    }
  });
};

var ResizeBehavior = function ResizeBehavior() {
  return _react.default.createElement(_HookshotTestTemplate.default, {
    id: "ResizeBehavior",
    type: "primary",
    hookshotContentProps: {
      closeOnEsc: false,
      closeOnOutsideClick: false,
      closeOnResize: true
    }
  });
};

var AllBehavior = function AllBehavior() {
  return _react.default.createElement(_HookshotTestTemplate.default, {
    id: "AllBehavior",
    type: "primary",
    hookshotContentProps: {
      closeOnEsc: true,
      closeOnOutsideClick: true,
      closeOnResize: true
    }
  });
};

var HookshotExample = function HookshotExample() {
  return _react.default.createElement("div", null, _react.default.createElement("p", null, "All Close Behavior"), _react.default.createElement(AllBehavior, null), _react.default.createElement("p", null, "Close on ESC Behavior"), _react.default.createElement(EscBehavior, null), _react.default.createElement("p", null, "Close on Outside Click Behavior"), _react.default.createElement(OutsideClickBehavior, null), _react.default.createElement("p", null, "Close on Outside Click Behavior With SVGs"), _react.default.createElement(OutsideClickBehaviorWithSvgs, null), _react.default.createElement("p", null, "Close on Resize Behavior"), _react.default.createElement(ResizeBehavior, null), _react.default.createElement("p", null, "No Close Behavior - Once opened, content stays on the screen when scrolling occurs."), _react.default.createElement(NoCloseBehavior, null), _react.default.createElement("button", {
    type: "button",
    id: "scroll-bounds"
  }, "Scroll Button"));
};

var _default = HookshotExample;
exports.default = _default;
