// terra-hookshot/lib/terra-dev-site/test/hookshot/HookshotEnabledBehaviors.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _HookshotTestTemplate = _interopRequireDefault(require("../common/HookshotTestTemplate"));

var Enabled = function Enabled() {
  return _react.default.createElement(_HookshotTestTemplate.default, {
    id: "Enabled",
    isOpen: true,
    type: "primary",
    isEnabled: true
  });
};

var NotEnabled = function NotEnabled() {
  return _react.default.createElement(_HookshotTestTemplate.default, {
    id: "NotEnabled",
    isOpen: true,
    type: "primary",
    isEnabled: false
  });
};

var HookshotExample = function HookshotExample() {
  return _react.default.createElement("div", {
    id: "EnabledBehaviors-bounds"
  }, _react.default.createElement("p", null, "Content Positioning is enabled. Content is visible."), _react.default.createElement(Enabled, null), _react.default.createElement("p", null, "Content Positioning is not enabled. Content is not visible."), _react.default.createElement(NotEnabled, null));
};

var _default = HookshotExample;
exports.default = _default;
