// terra-hookshot/lib/terra-dev-site/test/hookshot/HookshotAttachmentBehaviorNone.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _HookshotTestTemplate = _interopRequireDefault(require("../common/HookshotTestTemplate"));

var HookshotAttachmentBehaviorNone =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(HookshotAttachmentBehaviorNone, _React$Component);

  function HookshotAttachmentBehaviorNone(props) {
    var _this;

    (0, _classCallCheck2.default)(this, HookshotAttachmentBehaviorNone);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(HookshotAttachmentBehaviorNone).call(this, props));
    _this.handlePositionClick = _this.handlePositionClick.bind((0, _assertThisInitialized2.default)(_this));
    _this.state = {
      position: 'primary'
    };
    return _this;
  }

  (0, _createClass2.default)(HookshotAttachmentBehaviorNone, [{
    key: "handlePositionClick",
    value: function handlePositionClick(event) {
      this.setState({
        position: event.target.value
      });
    }
  }, {
    key: "render",
    value: function render() {
      var type = 'primary';

      if (this.state.position === 'offset') {
        type = 'none-offset';
      }

      return _react.default.createElement("div", null, _react.default.createElement(_HookshotTestTemplate.default, {
        attachmentBehavior: "none",
        id: "attachment-behavior-none",
        isOpen: false,
        type: type
      }), _react.default.createElement("p", null, " Choose the positioning behavior "), _react.default.createElement("p", null, " Primary position is on the middle right"), _react.default.createElement("button", {
        type: "button",
        id: "position-primary",
        value: "primary",
        onClick: this.handlePositionClick
      }, "Primary"), _react.default.createElement("button", {
        type: "button",
        id: "position-offset",
        value: "offset",
        onClick: this.handlePositionClick
      }, "Offset"));
    }
  }]);
  return HookshotAttachmentBehaviorNone;
}(_react.default.Component);

var _default = HookshotAttachmentBehaviorNone;
exports.default = _default;
