// terra-application-navigation/lib/common/_PopupMenuUser.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _terraAvatar = _interopRequireDefault(require("terra-avatar"));

var _propTypes = require("../utils/propTypes");

var _PopupMenuUserModule = _interopRequireDefault(require("./PopupMenuUser.module.scss"));

var cx = _bind.default.bind(_PopupMenuUserModule.default);

var propTypes = {
  /**
   * A configuration object with information pertaining to the application's user.
   */
  userConfig: _propTypes.userConfigPropType.isRequired
};

var PopupMenuUser = function PopupMenuUser(_ref) {
  var userConfig = _ref.userConfig;
  return _react.default.createElement("div", {
    className: cx('utility-user-layout')
  }, _react.default.createElement("div", {
    className: cx('avatar-container')
  }, _react.default.createElement("div", {
    className: cx('avatar-outline')
  }), _react.default.createElement(_terraAvatar.default, {
    alt: userConfig.name,
    image: userConfig.imageSrc,
    initials: userConfig.initials,
    size: "1.174rem"
  })), _react.default.createElement("div", {
    className: cx('info-container')
  }, _react.default.createElement("div", {
    "aria-hidden": true,
    className: cx('name')
  }, userConfig.name), userConfig.detail ? _react.default.createElement("div", {
    className: cx('detail')
  }, userConfig.detail) : null));
};

PopupMenuUser.propTypes = propTypes;
var _default = PopupMenuUser;
exports.default = _default;
