// terra-arrange/lib/terra-dev-site/test/arrange/AlignedAllLongWords.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Arrange = _interopRequireDefault(require("../../../Arrange"));

var _examplesetup = require("./common/examplesetup");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AlignedAllArrange = function AlignedAllArrange() {
  return _react.default.createElement("div", null, _react.default.createElement(_examplesetup.ArrangeWrapper, null, _react.default.createElement(_Arrange.default, {
    id: "default",
    fitStart: _examplesetup.longWordText,
    fill: _examplesetup.longWordText,
    fitEnd: _examplesetup.longWordText,
    fitStartAttributes: {
      className: 'test1',
      style: {
        width: '35%',
        wordWrap: 'break-word'
      }
    },
    fillAttributes: {
      className: 'test2',
      style: {
        width: '30%',
        wordWrap: 'break-word'
      }
    },
    fitEndAttributes: {
      className: 'test3',
      style: {
        width: '35%',
        wordWrap: 'break-word'
      }
    }
  })), _react.default.createElement(_examplesetup.ArrangeWrapper, null, _react.default.createElement(_Arrange.default, {
    id: "center",
    align: "center",
    fitStart: _examplesetup.longWordText,
    fill: _examplesetup.longWordText,
    fitEnd: _examplesetup.longWordText,
    fitStartAttributes: {
      className: 'test1',
      style: {
        width: '35%',
        wordWrap: 'break-word'
      }
    },
    fillAttributes: {
      className: 'test2',
      style: {
        width: '30%',
        wordWrap: 'break-word'
      }
    },
    fitEndAttributes: {
      className: 'test3',
      style: {
        width: '35%',
        wordWrap: 'break-word'
      }
    }
  })), _react.default.createElement(_examplesetup.ArrangeWrapper, null, _react.default.createElement(_Arrange.default, {
    id: "bottom",
    align: "bottom",
    fitStart: _examplesetup.longWordText,
    fill: _examplesetup.longWordText,
    fitEnd: _examplesetup.longWordText,
    fitStartAttributes: {
      className: 'test1',
      style: {
        width: '35%',
        wordWrap: 'break-word'
      }
    },
    fillAttributes: {
      className: 'test2',
      style: {
        width: '30%',
        wordWrap: 'break-word'
      }
    },
    fitEndAttributes: {
      className: 'test3',
      style: {
        width: '35%',
        wordWrap: 'break-word'
      }
    }
  })), _react.default.createElement(_examplesetup.ArrangeWrapper, null, _react.default.createElement(_Arrange.default, {
    id: "stretch",
    align: "stretch",
    fitStart: _examplesetup.longWordText,
    fill: _examplesetup.longWordText,
    fitEnd: _examplesetup.longWordText,
    fitStartAttributes: {
      className: 'test1',
      style: {
        width: '35%',
        wordWrap: 'break-word'
      }
    },
    fillAttributes: {
      className: 'test2',
      style: {
        width: '30%',
        wordWrap: 'break-word'
      }
    },
    fitEndAttributes: {
      className: 'test3',
      style: {
        width: '35%',
        wordWrap: 'break-word'
      }
    }
  })));
};

var _default = AlignedAllArrange;
exports.default = _default;
