// terra-arrange/lib/terra-dev-site/test/arrange/AlignedStretchMaxWidth.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Arrange = _interopRequireDefault(require("../../../Arrange"));

var _examplesetup = require("./common/examplesetup");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AlignedStretchMaxWidth = function AlignedStretchMaxWidth() {
  return _react.default.createElement("div", null, _react.default.createElement(_examplesetup.ArrangeWrapper, null, _react.default.createElement(_Arrange.default, {
    fitStart: _examplesetup.image,
    fill: _examplesetup.simpleText,
    fitEnd: _examplesetup.icon,
    alignFitStart: "stretch",
    alignFill: "stretch",
    alignFitEnd: "stretch",
    fitStartAttributes: {
      style: {
        maxWidth: '20%',
        wordWrap: 'break-word',
        overflow: 'hidden'
      }
    },
    fillAttributes: {
      style: {
        wordWrap: 'break-word',
        maxWidth: '20%'
      }
    },
    fitEndAttributes: {
      style: {
        maxWidth: '20%',
        wordWrap: 'break-word',
        overflow: 'hidden',
        backgroundColor: 'yellow'
      }
    }
  })));
};

var _default = AlignedStretchMaxWidth;
exports.default = _default;
