// terra-arrange/lib/terra-dev-site/test/arrange/IndividuallyAlignedFitStart.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Arrange = _interopRequireDefault(require("../../../Arrange"));

var _examplesetup = require("./common/examplesetup");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var IndividuallyAlignedFitStart = function IndividuallyAlignedFitStart() {
  return _react.default.createElement("div", null, _react.default.createElement(_examplesetup.ArrangeWrapper, null, _react.default.createElement(_Arrange.default, {
    id: "default",
    fitStart: _examplesetup.image,
    fill: _examplesetup.simpleText,
    fitEnd: _react.default.createElement(_examplesetup.IconWrapper, null)
  })), _react.default.createElement(_examplesetup.ArrangeWrapper, null, _react.default.createElement(_Arrange.default, {
    id: "center",
    alignFitStart: "center",
    fitStart: _examplesetup.image,
    fill: _examplesetup.simpleText,
    fitEnd: _react.default.createElement(_examplesetup.IconWrapper, null)
  })), _react.default.createElement(_examplesetup.ArrangeWrapper, null, _react.default.createElement(_Arrange.default, {
    id: "bottom",
    alignFitStart: "bottom",
    fitStart: _examplesetup.image,
    fill: _examplesetup.simpleText,
    fitEnd: _react.default.createElement(_examplesetup.IconWrapper, null)
  })), _react.default.createElement(_examplesetup.ArrangeWrapper, null, _react.default.createElement(_Arrange.default, {
    id: "stretch",
    alignFitStart: "stretch",
    fitStart: _examplesetup.image,
    fill: _examplesetup.simpleText,
    fitEnd: _react.default.createElement(_examplesetup.IconWrapper, null)
  })));
};

var _default = IndividuallyAlignedFitStart;
exports.default = _default;
