// terra-avatar/lib/terra-dev-site/doc/example/common/ColorVariantsWrapper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function ColorVariantsWrapper(WrappedComponent) {
  return (/*#__PURE__*/function (_React$Component) {
      _inherits(ColorVariant, _React$Component);

      function ColorVariant(props) {
        var _this;

        _classCallCheck(this, ColorVariant);

        _this = _possibleConstructorReturn(this, _getPrototypeOf(ColorVariant).call(this, props));
        _this.handleOnSelect = _this.handleOnSelect.bind(_assertThisInitialized(_this));
        _this.state = {
          colorVariant: 'auto'
        };
        return _this;
      }

      _createClass(ColorVariant, [{
        key: "handleOnSelect",
        value: function handleOnSelect(event) {
          this.setState(_defineProperty({}, event.target.name, event.target.value));
        }
      }, {
        key: "render",
        value: function render() {
          return _react.default.createElement("div", null, _react.default.createElement(WrappedComponent, _extends({
            color: this.state.colorVariant
          }, this.props)), _react.default.createElement("p", null, _react.default.createElement("label", {
            htmlFor: "colorVariant"
          }, "Select a color variant:")), _react.default.createElement("select", {
            id: "colorVariant",
            name: "colorVariant",
            value: this.state.colorVariant,
            onChange: this.handleOnSelect
          }, _react.default.createElement("option", {
            value: "auto"
          }, "Auto"), _react.default.createElement("option", {
            value: "neutral"
          }, "Neutral"), _react.default.createElement("option", {
            value: "one"
          }, "One"), _react.default.createElement("option", {
            value: "two"
          }, "Two"), _react.default.createElement("option", {
            value: "three"
          }, "Three"), _react.default.createElement("option", {
            value: "four"
          }, "Four"), _react.default.createElement("option", {
            value: "five"
          }, "Five"), _react.default.createElement("option", {
            value: "six"
          }, "Six"), _react.default.createElement("option", {
            value: "seven"
          }, "Seven"), _react.default.createElement("option", {
            value: "eight"
          }, "Eight"), _react.default.createElement("option", {
            value: "nine"
          }, "Nine"), _react.default.createElement("option", {
            value: "ten"
          }, "Ten")));
        }
      }]);

      return ColorVariant;
    }(_react.default.Component)
  );
}

var _default = ColorVariantsWrapper;
exports.default = _default;
