// terra-list/lib/List.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _ListModule = _interopRequireDefault(require("./List.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var cx = _bind.default.bind(_ListModule.default);

var propTypes = {
  /**
   * The children list items passed to the component.
   */
  children: _propTypes.default.node,

  /**
   * Whether or not the list's child items should have a border color applied.
   * One of `'none'`, `'standard'`, `'bottom-only'`.
   */
  dividerStyle: _propTypes.default.oneOf(['none', 'standard', 'bottom-only']),

  /**
   * The padding styling to apply to the child list item content.
   * One of `'none'`, `'standard'`, `'compact'`.
   */
  paddingStyle: _propTypes.default.oneOf(['none', 'standard', 'compact']),

  /**
   * Function callback for the ref of the ul.
   */
  refCallback: _propTypes.default.func,

  /**
   * Accessibility role of the list, defaults to 'none'. If creating a list with selectable items, pass 'listbox'.
   */
  role: _propTypes.default.string
};
var defaultProps = {
  children: [],
  dividerStyle: 'none',
  paddingStyle: 'none',
  role: 'none'
};

var List = function List(_ref) {
  var children = _ref.children,
      dividerStyle = _ref.dividerStyle,
      paddingStyle = _ref.paddingStyle,
      refCallback = _ref.refCallback,
      role = _ref.role,
      customProps = _objectWithoutProperties(_ref, ["children", "dividerStyle", "paddingStyle", "refCallback", "role"]);

  var listClassNames = cx(['list', {
    'padding-standard': paddingStyle === 'standard'
  }, {
    'padding-compact': paddingStyle === 'compact'
  }, {
    'divider-standard': dividerStyle === 'standard'
  }, {
    'divider-bottom-only': dividerStyle === 'bottom-only'
  }, customProps.className]);
  var attrSpread = {};

  if (role && role.length > 0 && role !== 'none') {
    attrSpread.role = role;
  }

  return _react.default.createElement("ul", _extends({}, customProps, attrSpread, {
    className: listClassNames,
    ref: refCallback
  }), children);
};

List.propTypes = propTypes;
List.defaultProps = defaultProps;
var _default = List;
exports.default = _default;
