// terra-arrange/lib/terra-dev-site/test/arrange/common/examplesetup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.longWordText = exports.IconWrapper = exports.ArrangeWrapper = exports.longText = exports.simpleText = exports.image = exports.icon = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _IconAlert = _interopRequireDefault(require("terra-icon/lib/icon/IconAlert"));

var _examplesetuptest = _interopRequireDefault(require("./examplesetuptest.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_examplesetuptest.default);

var icon = _react.default.createElement(_IconAlert.default, {
  width: "4em",
  height: "4em"
}); // Block style is required since images are inline by default. This could be resolved by someones normalize.css


exports.icon = icon;

var image = _react.default.createElement("img", {
  className: cx('image-wrapper'),
  height: "75",
  width: "75",
  src: "https://github.com/cerner/terra-core/raw/master/terra.png",
  alt: "a happy panda"
});

exports.image = image;
var ipsum = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.';
var longWord = 'Thisisareallyreallylongwordthathasnospacesthatwillneedtobebrokeninordertohavethetextwrapandnotoverlowintothenextelement';

var simpleText = _react.default.createElement("div", null, ipsum);

exports.simpleText = simpleText;

var longText = _react.default.createElement("div", null, ipsum, ' ', ipsum, ' ', ipsum, ' ', ipsum);

exports.longText = longText;

var longWordText = _react.default.createElement("div", null, longWord);

exports.longWordText = longWordText;

var ArrangeWrapper = function ArrangeWrapper(props) {
  return _react.default.createElement("div", {
    className: cx('arrange-wrapper')
  }, props.children);
};

exports.ArrangeWrapper = ArrangeWrapper;
ArrangeWrapper.propTypes = {
  children: _propTypes.default.element
};

var IconWrapper = function IconWrapper() {
  return _react.default.createElement("div", {
    className: cx('icon-wrapper')
  }, _react.default.createElement("svg", {
    className: "terra-Icon",
    height: "4em",
    width: "4em",
    viewBox: "0 0 48 48"
  }, _react.default.createElement("path", {
    d: "M24 0h-.1C10.7 0 0 10.8 0 24c0 13.3 10.7 24 24 24s24-10.7 24-24S37.3 0 24 0zm-4 36.4L6.7 23.1l3.6-3.6 9.7 9.9 17.7-17.9 3.6 3.6L20 36.4z",
    fill: "#FFF"
  })));
};

exports.IconWrapper = IconWrapper;
