// terra-button/lib/terra-dev-site/test/button/Variants/ActionButton.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconSquareSymbol = _interopRequireDefault(require("terra-icon/lib/icon/IconSquareSymbol"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Button = _interopRequireDefault(require("../../../../Button"));

var _ButtonTestCommonModule = _interopRequireDefault(require("../ButtonTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ButtonTestCommonModule.default);

var ActionButton = function ActionButton() {
  return _react.default.createElement("div", {
    id: "action",
    className: cx('button-wrapper')
  }, _react.default.createElement(_Button.default, {
    id: "actionButton",
    text: "Action",
    icon: _react.default.createElement(_IconSquareSymbol.default, null),
    variant: "action"
  }), _react.default.createElement(_Button.default, {
    id: "actionButtonDisabled",
    text: "Action Disabled",
    icon: _react.default.createElement(_IconSquareSymbol.default, null),
    variant: "action",
    isDisabled: true
  }));
};

var _default = ActionButton;
exports.default = _default;
