// terra-button/lib/terra-dev-site/test/button/Variants/DeemphasisButton.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Button = _interopRequireDefault(require("../../../../Button"));

var _ButtonTestCommonModule = _interopRequireDefault(require("../ButtonTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ButtonTestCommonModule.default);

var DeemphasisButton = function DeemphasisButton() {
  return _react.default.createElement("div", {
    id: "de-emphasis",
    className: cx('button-wrapper')
  }, _react.default.createElement(_Button.default, {
    id: "de-emphasisButton",
    text: "De-emphasis",
    variant: "de-emphasis"
  }), _react.default.createElement(_Button.default, {
    id: "de-emphasisButtonDisabled",
    text: "De-emphasis Disabled",
    variant: "de-emphasis",
    isDisabled: true
  }));
};

var _default = DeemphasisButton;
exports.default = _default;
