// terra-button/lib/terra-dev-site/test/button/Variants/GhostButton.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Button = _interopRequireDefault(require("../../../../Button"));

var _ButtonTestCommonModule = _interopRequireDefault(require("../ButtonTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ButtonTestCommonModule.default);

var GhostButton = function GhostButton() {
  return _react.default.createElement("div", {
    id: "ghost",
    className: cx('button-wrapper')
  }, _react.default.createElement(_Button.default, {
    id: "ghostButton",
    text: "Ghost",
    variant: "ghost"
  }), _react.default.createElement(_Button.default, {
    id: "ghostButtonDisabled",
    text: "Ghost Disabled",
    variant: "ghost",
    isDisabled: true
  }));
};

var _default = GhostButton;
exports.default = _default;
