// terra-button/lib/terra-dev-site/test/button/Variants/NeutralButton.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Button = _interopRequireDefault(require("../../../../Button"));

var _ButtonTestCommonModule = _interopRequireDefault(require("../ButtonTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ButtonTestCommonModule.default);

var NeutralButton = function NeutralButton() {
  return _react.default.createElement("div", {
    id: "neutral",
    className: cx('button-wrapper')
  }, _react.default.createElement(_Button.default, {
    id: "neutralButton",
    text: "Neutral",
    variant: "neutral"
  }), _react.default.createElement(_Button.default, {
    id: "neutralButtonDisabled",
    text: "Neutral Disabled",
    variant: "neutral",
    isDisabled: true
  }));
};

var _default = NeutralButton;
exports.default = _default;
