// terra-slide-panel-manager/lib/terra-dev-site/test/slide-panel-manager/SlidePanelManagerWithDisclosureAccessory.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _DisclosureComponent = _interopRequireDefault(require("./DisclosureComponent"));

var _SlidePanelManager = _interopRequireDefault(require("../../../SlidePanelManager"));

var _SlidePanelManagerDefaultTestModule = _interopRequireDefault(require("./SlidePanelManagerDefault.test.module.scss"));

var cx = _bind.default.bind(_SlidePanelManagerDefaultTestModule.default);

var SlidePanelManagerWithDisclosureAccessory = function SlidePanelManagerWithDisclosureAccessory() {
  return _react.default.createElement("div", {
    role: "main",
    className: cx('content-wrapper')
  }, _react.default.createElement(_SlidePanelManager.default, {
    disclosureAccessory: _react.default.createElement("div", {
      className: cx('disclosure-accessory')
    }, "Disclosure Accessory")
  }, _react.default.createElement(_DisclosureComponent.default, {
    identifier: "root-component",
    disclosureType: "panel",
    renderHeaderAdapter: true
  })));
};

var _default = SlidePanelManagerWithDisclosureAccessory;
exports.default = _default;
