// terra-notification-dialog/lib/terra-dev-site/test/notification-dialog/CompleteNotificationDialogWithLongMessage.test.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _terraButton = _interopRequireDefault(require("terra-button"));

var _NotificationDialog = _interopRequireWildcard(require("../../../NotificationDialog"));

var clickConfirm = function clickConfirm() {
  alert('You clicked confirm'); // eslint-disable-line no-alert
};

var CompleteNotificationDialogWithLongMessage = function CompleteNotificationDialogWithLongMessage() {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      isOpen = _useState2[0],
      setIsOpen = _useState2[1];

  var handleOpenModal = function handleOpenModal() {
    setIsOpen(true);
  };

  var handleCloseModal = function handleCloseModal() {
    setIsOpen(false);
  };

  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_NotificationDialog.default, {
    variant: _NotificationDialog.NotificationDialogVariants.SUCCESS,
    isOpen: isOpen,
    title: "This is the titleThis is the titleThis is the titleThis is the titleThis is the titleThis is the titleThis is the titleThis is the title",
    startMessage: "This is the messageThis is the messageThis is the messageThis is the messageThis is the messageThis is the messageThis is the messageThis is the messageThis is the messageThis is the message",
    acceptAction: {
      text: 'Confirm',
      onClick: clickConfirm
    },
    rejectAction: {
      text: 'Close',
      onClick: handleCloseModal
    },
    buttonOrder: "acceptFirst",
    emphasizedAction: "accept"
  }), _react.default.createElement(_terraButton.default, {
    id: "trigger-notification-dialog",
    text: "Trigger NotificationDialog",
    onClick: handleOpenModal
  }));
};

var _default = CompleteNotificationDialogWithLongMessage;
exports.default = _default;
