// terra-notification-dialog/lib/terra-dev-site/test/notification-dialog/NoVariantNotificationDialog.test.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _terraButton = _interopRequireDefault(require("terra-button"));

var _NotificationDialog = _interopRequireDefault(require("../../../NotificationDialog"));

var clickConfirm = function clickConfirm() {
  alert('You clicked confirm'); // eslint-disable-line no-alert
};

var NoVariantNotificationDialog = function NoVariantNotificationDialog() {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      isOpen = _useState2[0],
      setIsOpen = _useState2[1];

  var handleOpenModal = function handleOpenModal() {
    setIsOpen(true);
  };

  var handleCloseModal = function handleCloseModal() {
    setIsOpen(false);
  };

  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_NotificationDialog.default, {
    header: "No Variant for this dialog",
    isOpen: isOpen,
    title: "Make sure that the title relates directly to the choices.",
    startMessage: "The Main Instruction is text used to provide more detail or define terminology. Don\u2019t repeat the title verbatim.",
    acceptAction: {
      text: 'Confirm',
      onClick: clickConfirm
    },
    rejectAction: {
      text: 'Close',
      onClick: handleCloseModal
    },
    buttonOrder: "acceptFirst",
    emphasizedAction: "accept"
  }), _react.default.createElement(_terraButton.default, {
    id: "trigger-notification-dialog",
    text: "Trigger Notification Dialog",
    onClick: handleOpenModal
  }));
};

var _default = NoVariantNotificationDialog;
exports.default = _default;
