// terra-form-input/lib/terra-dev-site/test/form-input/DisabledInputField.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _InputField = _interopRequireDefault(require("../../../InputField"));

var _InputTestModule = _interopRequireDefault(require("./common/Input.test.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_InputTestModule.default);

var DisabledInputFieldExample = function DisabledInputFieldExample() {
  return _react.default.createElement("div", {
    className: cx('content-wrapper')
  }, _react.default.createElement(_InputField.default, {
    disabled: true,
    inputId: "test-input",
    defaultValue: "Value",
    error: "Text",
    help: "Help",
    hideRequired: true,
    inputAttrs: {
      name: 'test',
      type: 'text'
    },
    label: "Label Text",
    labelAttrs: {
      className: 'label'
    },
    showOptional: true,
    maxWidth: "40%"
  }));
};

var _default = DisabledInputFieldExample;
exports.default = _default;
