// terra-form-input/lib/terra-dev-site/test/form-input/InputField.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _InputField = _interopRequireDefault(require("../../../InputField"));

var _InputTestModule = _interopRequireDefault(require("./common/Input.test.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var cx = _bind.default.bind(_InputTestModule.default);

var InputFieldExample = /*#__PURE__*/function (_React$Component) {
  _inherits(InputFieldExample, _React$Component);

  function InputFieldExample() {
    var _this;

    _classCallCheck(this, InputFieldExample);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(InputFieldExample).call(this));
    _this.state = {
      isInvalid: false,
      isIncomplete: false,
      required: false
    };
    _this.toggleInvalid = _this.toggleInvalid.bind(_assertThisInitialized(_this));
    _this.toggleIncomplete = _this.toggleIncomplete.bind(_assertThisInitialized(_this));
    return _this;
  }

  _createClass(InputFieldExample, [{
    key: "toggleInvalid",
    value: function toggleInvalid() {
      this.setState(function (prevState) {
        return {
          isInvalid: !prevState.isInvalid
        };
      });
    }
  }, {
    key: "toggleIncomplete",
    value: function toggleIncomplete() {
      this.setState(function (prevState) {
        return {
          isIncomplete: !prevState.isIncomplete,
          required: !prevState.required
        };
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", {
        className: cx('content-wrapper')
      }, _react.default.createElement("button", {
        type: "button",
        id: "validity-toggle",
        onClick: this.toggleInvalid
      }, "Toggle Validity"), _react.default.createElement("button", {
        type: "button",
        id: "incomplete-toggle",
        onClick: this.toggleIncomplete
      }, "Toggle Incomplete"), _react.default.createElement(_InputField.default, {
        inputId: "test-input",
        defaultValue: "Value",
        error: "Error message.",
        help: "Help message.",
        hideRequired: true,
        inputAttrs: {
          name: 'test',
          type: 'text'
        },
        isInvalid: this.state.isInvalid,
        isIncomplete: this.state.isIncomplete,
        required: this.state.required,
        label: "Label Text",
        labelAttrs: {
          className: 'label'
        },
        showOptional: true
      }));
    }
  }]);

  return InputFieldExample;
}(_react.default.Component);

var _default = InputFieldExample;
exports.default = _default;
