// terra-status-view/lib/terra-dev-site/test/status-view/StatusViewThinConstraint.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _StatusView = _interopRequireDefault(require("../../../StatusView"));

var _StatusViewTestCommonModule = _interopRequireDefault(require("./StatusViewTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_StatusViewTestCommonModule.default);

var ipsum = 'Lorem ipsum dolor';
var StatusViewButtons = [{
  text: ipsum,
  key: 1,
  id: 'button1'
}, {
  text: ipsum,
  key: 2,
  id: 'button2'
}];
/* eslint-disable react/forbid-dom-props */

var _default = function _default() {
  return _react.default.createElement("div", {
    className: cx('status-view-wrapper-2')
  }, _react.default.createElement(_StatusView.default, {
    id: "statusView",
    variant: "error",
    title: ipsum,
    message: ipsum,
    buttonAttrs: StatusViewButtons
  }));
};

exports.default = _default;
