// terra-arrange/lib/terra-dev-site/doc/example/ArrangeAlignAllContainers.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _terraArrange = _interopRequireDefault(require("terra-arrange"));

var _examplesetup = require("../common/examplesetup");

var _ArrangeExamplesModule = _interopRequireDefault(require("./ArrangeExamples.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ArrangeExamplesModule.default);

var ArrangeAlignAllContainers = function ArrangeAlignAllContainers() {
  return _react.default.createElement("div", null, _react.default.createElement("h3", null, "Align - Default"), _react.default.createElement(_terraArrange.default, {
    className: cx('arrange'),
    fitStart: _examplesetup.alignExampleDivBlue,
    fill: _examplesetup.textWithBlueBorder,
    fitEnd: _examplesetup.alignExampleDivBlue
  }), _react.default.createElement("br", null), _react.default.createElement("hr", null), _react.default.createElement("h3", null, "Align - Center"), _react.default.createElement(_terraArrange.default, {
    align: "center",
    className: cx('arrange'),
    fitStart: _examplesetup.alignExampleDivBlue,
    fill: _examplesetup.textWithBlueBorder,
    fitEnd: _examplesetup.alignExampleDivBlue
  }), _react.default.createElement("br", null), _react.default.createElement("hr", null), _react.default.createElement("h3", null, "Align - Bottom"), _react.default.createElement(_terraArrange.default, {
    align: "bottom",
    className: cx('arrange'),
    fitStart: _examplesetup.alignExampleDivBlue,
    fill: _examplesetup.textWithBlueBorder,
    fitEnd: _examplesetup.alignExampleDivBlue
  }));
};

var _default = ArrangeAlignAllContainers;
exports.default = _default;
