// terra-form-select/lib/TagSelect.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactIntl = require("react-intl");

var _Frame = _interopRequireDefault(require("./tag/Frame"));

var _Option = _interopRequireDefault(require("./shared/_Option"));

var _OptGroup = _interopRequireDefault(require("./shared/_OptGroup"));

var _Tag = _interopRequireDefault(require("./shared/_Tag"));

var _SelectUtil = _interopRequireDefault(require("./shared/_SelectUtil"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance"); }

function _iterableToArray(iter) { if (Symbol.iterator in Object(iter) || Object.prototype.toString.call(iter) === "[object Arguments]") return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = new Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var propTypes = {
  /**
   * The dropdown menu options.
   */
  children: _propTypes.default.node,

  /**
   * The default selected value. Can be a string, number, or array of strings/numbers.
   */
  defaultValue: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number, _propTypes.default.array]),

  /**
   * Whether the select is disabled.
   */
  disabled: _propTypes.default.bool,

  /**
   * Additional attributes to spread onto the dropdownq. ( Style, ClassNames, etc.. )
   */
  // eslint-disable-next-line react/forbid-prop-types
  dropdownAttrs: _propTypes.default.object,

  /**
   * @private
   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
   */
  intl: _reactIntl.intlShape.isRequired,

  /**
   * Whether the select displays as Incomplete. Use when no value has been provided. _(usage note: `required` must also be set)_.
   */
  isIncomplete: _propTypes.default.bool,

  /**
   * Whether the select displays as Invalid. Use when value does not meet validation pattern.
   */
  isInvalid: _propTypes.default.bool,

  /**
   * Ensure accessibility on touch devices. Will render the dropdown menu in
   * normal DOM flow with position absolute. By default, the menu renders in a
   * portal, which is inaccessible on touch devices.
   */
  isTouchAccessible: _propTypes.default.bool,

  /**
   * The max height of the dropdown.
   */
  maxHeight: _propTypes.default.number,

  /**
   * @private The maximum number of options that can be selected. A value less than 2 will be ignored.
   */
  maxSelectionCount: _propTypes.default.number,

  /**
   * Callback function triggered when the select loses focus. function(event)
   */
  onBlur: _propTypes.default.func,

  /**
   * Callback function triggered when the value changes. function(value)
   */
  onChange: _propTypes.default.func,

  /**
   * Callback function triggered when the select is clicked. function(event)
   */
  onClick: _propTypes.default.func,

  /**
   * Callback function triggered when an option is deselected. function(value)
   */
  onDeselect: _propTypes.default.func,

  /**
   * Callback function triggered when the select receives focus. function(event)
   */
  onFocus: _propTypes.default.func,

  /**
   * Callback function triggered when the search criteria changes. function(searchValue)
   */
  onSearch: _propTypes.default.func,

  /**
   * Callback function triggered when an option is selected. function(value)
   */
  onSelect: _propTypes.default.func,

  /**
   * Callback function invoked for each option on search change. function(searchValue, option)
   */
  optionFilter: _propTypes.default.func,

  /**
   * Placeholder text.
   */
  placeholder: _propTypes.default.string,

  /**
   * Whether the field is required.
   */
  required: _propTypes.default.bool,

  /**
   * The selected value. Can be a string, number, or array of strings/numbers.
   */
  value: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number, _propTypes.default.array]),

  /**
   * @private
   * The id of the input field.
   */
  inputId: _propTypes.default.string
};
var defaultProps = {
  children: undefined,
  defaultValue: undefined,
  disabled: false,
  dropdownAttrs: undefined,
  isIncomplete: false,
  isInvalid: false,
  isTouchAccessible: false,
  maxSelectionCount: undefined,
  onChange: undefined,
  onDeselect: undefined,
  onSearch: undefined,
  onSelect: undefined,
  optionFilter: undefined,
  placeholder: undefined,
  required: false,
  value: undefined,
  inputId: undefined
};

var TagSelect = /*#__PURE__*/function (_React$Component) {
  _inherits(TagSelect, _React$Component);

  function TagSelect(props) {
    var _this;

    _classCallCheck(this, TagSelect);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(TagSelect).call(this, props));
    var defaultValue = props.defaultValue,
        value = props.value;
    _this.state = {
      tags: [],
      value: _SelectUtil.default.defaultValue({
        defaultValue: defaultValue,
        value: value,
        multiple: true
      })
    };
    _this.display = _this.display.bind(_assertThisInitialized(_this));
    _this.handleChange = _this.handleChange.bind(_assertThisInitialized(_this));
    _this.handleDeselect = _this.handleDeselect.bind(_assertThisInitialized(_this));
    _this.handleSelect = _this.handleSelect.bind(_assertThisInitialized(_this));
    return _this;
  }
  /**
   * Returns the appropriate variant display
   */


  _createClass(TagSelect, [{
    key: "display",
    value: function display() {
      var _this2 = this;

      var selectValue = _SelectUtil.default.value(this.props, this.state);

      return selectValue.map(function (tag) {
        return _react.default.createElement(_Tag.default, {
          value: tag,
          key: tag,
          onDeselect: _this2.handleDeselect
        }, _SelectUtil.default.valueDisplay(_this2.props, tag));
      });
    }
    /**
     * Communicates changes to the value.
     * @param {array|number|string} value - The value resulting from a change.
     */

  }, {
    key: "handleChange",
    value: function handleChange(value) {
      if (this.props.value === undefined) {
        this.setState({
          value: value
        });
      }

      if (this.props.onChange) {
        this.props.onChange(value);
      }
    }
    /**
     * Communicates the removal of a value from the selected options.
     * @param {number|string} value - The value to be removed.
     */

  }, {
    key: "handleDeselect",
    value: function handleDeselect(value) {
      this.handleChange(_SelectUtil.default.deselect(this.props, this.state, value));

      if (this.props.onDeselect) {
        this.props.onDeselect(value);
      }
    }
    /**
     * Communicates the selection of a value.
     * @param {number|string} value - The value of the selected option.
     * @param {ReactNode} option - The selected option.
     */

  }, {
    key: "handleSelect",
    value: function handleSelect(value, option) {
      var newValue = [].concat(_toConsumableArray(_SelectUtil.default.value(this.props, this.state)), [value]);
      this.handleChange(newValue); // Add new tags for uncontrolled components.

      if (this.props.value === undefined && !_SelectUtil.default.findByValue(this.props, this.state, value)) {
        this.setState(function (prevState) {
          return {
            tags: [].concat(_toConsumableArray(prevState.tags), [_react.default.createElement(_Option.default, {
              key: value,
              display: value,
              value: value
            })])
          };
        });
      }

      if (this.props.onSelect) {
        this.props.onSelect(value, option);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          defaultValue = _this$props.defaultValue,
          onChange = _this$props.onChange,
          placeholder = _this$props.placeholder,
          required = _this$props.required,
          value = _this$props.value,
          intl = _this$props.intl,
          inputId = _this$props.inputId,
          otherProps = _objectWithoutProperties(_this$props, ["children", "defaultValue", "onChange", "placeholder", "required", "value", "intl", "inputId"]);

      var defaultPlaceholder = intl.formatMessage({
        id: 'Terra.form.select.defaultDisplay'
      });
      var selectPlaceholder = placeholder === undefined ? defaultPlaceholder : placeholder;
      return _react.default.createElement(_Frame.default, _extends({}, otherProps, {
        "data-terra-select": true,
        value: _SelectUtil.default.value(this.props, this.state),
        display: this.display(),
        onDeselect: this.handleDeselect,
        onSelect: this.handleSelect,
        placeholder: selectPlaceholder,
        required: required,
        totalOptions: _SelectUtil.default.getTotalNumberOfOptions(children),
        inputId: inputId
      }), this.state.tags, children);
    }
  }]);

  return TagSelect;
}(_react.default.Component);

TagSelect.Option = _Option.default;
TagSelect.OptGroup = _OptGroup.default;
TagSelect.propTypes = propTypes;
TagSelect.defaultProps = defaultProps;
TagSelect.isSelect = true;

var _default = (0, _reactIntl.injectIntl)(TagSelect);

exports.default = _default;
