// terra-spacer/lib/terra-dev-site/test/spacer/SpacerLarge+4.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Spacer = _interopRequireDefault(require("../../../Spacer"));

var _SpacerTestModule = _interopRequireDefault(require("./SpacerTest.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_SpacerTestModule.default);

var _default = function _default() {
  return _react.default.createElement("div", null, _react.default.createElement("h3", null, "Background color added for context. The light orange background shows margin, the green background shows padding."), _react.default.createElement("div", null, "Margin and Padding with value large+4 applied"), _react.default.createElement("div", {
    className: cx(['spacing-demo', 'spacer-demo-margin'])
  }, _react.default.createElement(_Spacer.default, {
    id: "spacerMarginPaddingLarge4",
    className: cx('spacer'),
    marginTop: "large+4",
    marginBottom: "large+4",
    marginLeft: "large+4",
    marginRight: "large+4",
    paddingTop: "large+4",
    paddingBottom: "large+4",
    paddingLeft: "large+4",
    paddingRight: "large+4"
  }, _react.default.createElement("div", {
    className: cx('spacer-content-wrapper')
  }, "Test component"))), _react.default.createElement("div", null, "Margin Padding with value large+4 and isInlineBlock prop applied"), _react.default.createElement("div", {
    className: cx(['spacing-demo', 'spacer-demo-margin'])
  }, _react.default.createElement(_Spacer.default, {
    id: "spacerMarginPaddingLarge4InlineBlock",
    className: cx('spacer'),
    marginTop: "large+4",
    marginBottom: "large+4",
    marginLeft: "large+4",
    marginRight: "large+4",
    paddingTop: "large+4",
    paddingBottom: "large+4",
    paddingLeft: "large+4",
    paddingRight: "large+4",
    isInlineBlock: true
  }, _react.default.createElement("div", {
    className: cx('spacer-content-wrapper')
  }, "Test component"))));
};

exports.default = _default;
