// terra-spacer/lib/terra-dev-site/test/spacer/SpacerSmall-1.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Spacer = _interopRequireDefault(require("../../../Spacer"));

var _SpacerTestModule = _interopRequireDefault(require("./SpacerTest.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_SpacerTestModule.default);

var _default = function _default() {
  return _react.default.createElement("div", null, _react.default.createElement("h3", null, "Background color added for context. The light orange background shows margin, the green background shows padding."), _react.default.createElement("div", null, "Margin with value small-1 applied"), _react.default.createElement("div", {
    className: cx(['spacing-demo', 'spacer-demo-margin'])
  }, _react.default.createElement(_Spacer.default, {
    id: "spacerMarginSmall1",
    className: cx('spacer'),
    marginTop: "small-1",
    marginBottom: "small-1",
    marginLeft: "small-1",
    marginRight: "small-1"
  }, _react.default.createElement("div", {
    className: cx('spacer-content-wrapper')
  }, "Test component"))), _react.default.createElement("div", null, "Padding with value small-1 applied"), _react.default.createElement("div", {
    className: cx(['spacing-demo', 'spacer-demo-margin'])
  }, _react.default.createElement(_Spacer.default, {
    id: "spacerPaddingSmall1",
    className: cx('spacer'),
    paddingTop: "small-1",
    paddingBottom: "small-1",
    paddingLeft: "small-1",
    paddingRight: "small-1"
  }, _react.default.createElement("div", {
    className: cx('spacer-content-wrapper')
  }, "Test component"))), _react.default.createElement("div", null, "Margin and Padding with value small-1 applied"), _react.default.createElement("div", {
    className: cx(['spacing-demo', 'spacer-demo-margin'])
  }, _react.default.createElement(_Spacer.default, {
    id: "spacerMarginPaddingSmall1",
    className: cx('spacer'),
    marginTop: "small-1",
    marginBottom: "small-1",
    marginLeft: "small-1",
    marginRight: "small-1",
    paddingTop: "small-1",
    paddingBottom: "small-1",
    paddingLeft: "small-1",
    paddingRight: "small-1"
  }, _react.default.createElement("div", {
    className: cx('spacer-content-wrapper')
  }, "Test component"))), _react.default.createElement("div", null, "Margin, Padding with value small-1 and isInlineBlock prop applied"), _react.default.createElement("div", {
    className: cx(['spacing-demo', 'spacer-demo-margin'])
  }, _react.default.createElement(_Spacer.default, {
    id: "spacerMarginPaddingSmall1InlineBlock",
    className: cx('spacer'),
    marginTop: "small-1",
    marginBottom: "small-1",
    marginLeft: "small-1",
    marginRight: "small-1",
    paddingTop: "small-1",
    paddingBottom: "small-1",
    paddingLeft: "small-1",
    paddingRight: "small-1",
    isInlineBlock: true
  }, _react.default.createElement("div", {
    className: cx('spacer-content-wrapper')
  }, "Test component"))));
};

exports.default = _default;
