// terra-spacer/lib/terra-dev-site/test/spacer/SpacerMedium.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Spacer = _interopRequireDefault(require("../../../Spacer"));

var _SpacerTestModule = _interopRequireDefault(require("./SpacerTest.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_SpacerTestModule.default);

var _default = function _default() {
  return _react.default.createElement("div", null, _react.default.createElement("h3", null, "Background color added for context. The light orange background shows margin, the green background shows padding."), _react.default.createElement("div", null, "Margin with value medium applied"), _react.default.createElement("div", {
    className: cx(['spacing-demo', 'spacer-demo-margin'])
  }, _react.default.createElement(_Spacer.default, {
    id: "spacerMarginMedium",
    className: cx('spacer'),
    marginTop: "medium",
    marginBottom: "medium",
    marginLeft: "medium",
    marginRight: "medium"
  }, _react.default.createElement("div", {
    className: cx('spacer-content-wrapper')
  }, "Test component"))), _react.default.createElement("div", null, "Padding with value medium applied"), _react.default.createElement("div", {
    className: cx(['spacing-demo', 'spacer-demo-margin'])
  }, _react.default.createElement(_Spacer.default, {
    id: "spacerPaddingMedium",
    className: cx('spacer'),
    paddingTop: "medium",
    paddingBottom: "medium",
    paddingLeft: "medium",
    paddingRight: "medium"
  }, _react.default.createElement("div", {
    className: cx('spacer-content-wrapper')
  }, "Test component"))), _react.default.createElement("div", null, "Margin and Padding with value medium applied"), _react.default.createElement("div", {
    className: cx(['spacing-demo', 'spacer-demo-margin'])
  }, _react.default.createElement(_Spacer.default, {
    id: "spacerMarginPaddingMedium",
    className: cx('spacer'),
    marginTop: "medium",
    marginBottom: "medium",
    marginLeft: "medium",
    marginRight: "medium",
    paddingTop: "medium",
    paddingBottom: "medium",
    paddingLeft: "medium",
    paddingRight: "medium"
  }, _react.default.createElement("div", {
    className: cx('spacer-content-wrapper')
  }, "Test component"))), _react.default.createElement("div", null, "Margin Padding with value medium and isInlineBlock prop applied"), _react.default.createElement("div", {
    className: cx(['spacing-demo', 'spacer-demo-margin'])
  }, _react.default.createElement(_Spacer.default, {
    id: "spacerMarginPaddingMediumInlineBlock",
    className: cx('spacer'),
    marginTop: "medium",
    marginBottom: "medium",
    marginLeft: "medium",
    marginRight: "medium",
    paddingTop: "medium",
    paddingBottom: "medium",
    paddingLeft: "medium",
    paddingRight: "medium",
    isInlineBlock: true
  }, _react.default.createElement("div", {
    className: cx('spacer-content-wrapper')
  }, "Test component"))));
};

exports.default = _default;
