// terra-action-footer/lib/terra-dev-site/doc/example/standard/MultipleStartEndActions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraButton = _interopRequireDefault(require("terra-button"));

var _terraHyperlink = _interopRequireDefault(require("terra-hyperlink"));

var _terraSpacer = _interopRequireDefault(require("terra-spacer"));

var _terraActionFooter = _interopRequireDefault(require("terra-action-footer"));

var _ExampleTemplate = _interopRequireDefault(require("../../common/ExampleTemplate"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = function _default() {
  return _react.default.createElement(_ExampleTemplate.default, null, _react.default.createElement(_terraActionFooter.default, {
    start: _react.default.createElement(_terraHyperlink.default, {
      href: "#"
    }, "Start Action"),
    end: _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_terraSpacer.default, {
      isInlineBlock: true,
      marginRight: "medium"
    }, _react.default.createElement(_terraButton.default, {
      text: "Submit",
      variant: _terraButton.default.Opts.Variants.EMPHASIS
    })), _react.default.createElement(_terraButton.default, {
      text: "Cancel"
    }))
  }));
};

exports.default = _default;
