// terra-button/lib/terra-dev-site/test/button/ButtonTypes.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Button = _interopRequireDefault(require("../../../Button"));

var _ButtonTestCommonModule = _interopRequireDefault(require("./ButtonTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ButtonTestCommonModule.default);

var ButtonWithTypeButton = function ButtonWithTypeButton() {
  return _react.default.createElement(_Button.default, {
    id: "buttonWithTypeButton",
    text: "Button Type is Default",
    type: "button"
  });
};

var ButtonWithTypeReset = function ButtonWithTypeReset() {
  return _react.default.createElement(_Button.default, {
    id: "buttonWithTypeReset",
    text: "Button Type is Reset",
    type: "reset"
  });
};

var ButtonWithTypeSubmit = function ButtonWithTypeSubmit() {
  return _react.default.createElement(_Button.default, {
    id: "buttonWithTypeSubmit",
    text: "Button Type is Submit",
    type: "submit"
  });
};

var _default = function _default() {
  return _react.default.createElement("div", null, _react.default.createElement(ButtonWithTypeButton, null), _react.default.createElement("div", {
    className: cx('spacing')
  }), _react.default.createElement(ButtonWithTypeReset, null), _react.default.createElement("div", {
    className: cx('spacing')
  }), _react.default.createElement(ButtonWithTypeSubmit, null));
};

exports.default = _default;
