// terra-button/lib/terra-dev-site/test/button/LongTextButton.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconSquareSymbol = _interopRequireDefault(require("terra-icon/lib/icon/IconSquareSymbol"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Button = _interopRequireDefault(require("../../../Button"));

var _ButtonTestCommonModule = _interopRequireDefault(require("./ButtonTestCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ButtonTestCommonModule.default);

var text = 'Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long';
var words = 'Art artisan blog future next level odd park party polaroid truffaut. Apparel banksy beard belly bespoke biodiesel craft dreamcatcher ethical high pork quinoa retro semiotics squid tumblr. AnIncrediblyIncrediblyIncrediblyIncrediblyIncrediblyIncrediblyIncrediblyIncrediblyIncrediblyIncrediblyIncrediblyLongWord.';

var LongTextButton = function LongTextButton() {
  return _react.default.createElement(_Button.default, {
    id: "long-text",
    text: text
  });
};

var LongTextBlockButton = function LongTextBlockButton() {
  return _react.default.createElement(_Button.default, {
    id: "long-text-block",
    text: text,
    isBlock: true
  });
};

var _default = function _default() {
  return _react.default.createElement("div", null, _react.default.createElement(LongTextButton, null), _react.default.createElement("div", {
    className: cx('spacing')
  }), _react.default.createElement(LongTextBlockButton, null), _react.default.createElement("div", {
    className: cx('spacing')
  }), _react.default.createElement(_Button.default, {
    id: "long-word",
    text: words
  }), _react.default.createElement("div", {
    className: cx('spacing')
  }), _react.default.createElement(_Button.default, {
    id: "long-word-block",
    text: words,
    isBlock: true
  }), _react.default.createElement("div", {
    className: cx('spacing')
  }), _react.default.createElement(_Button.default, {
    icon: _react.default.createElement(_IconSquareSymbol.default, null),
    text: words
  }), _react.default.createElement("div", {
    className: cx('spacing')
  }), _react.default.createElement(_Button.default, {
    icon: _react.default.createElement(_IconSquareSymbol.default, null),
    text: words,
    isReversed: true
  }));
};

exports.default = _default;
