// terra-slide-panel/lib/terra-dev-site/test/slide-panel/SlidePanelToggle.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _SlidePanel = _interopRequireDefault(require("../../../SlidePanel"));

var _SlidePanelDocCommonTestModule = _interopRequireDefault(require("./SlidePanelDocCommon.test.module.scss"));

var cx = _bind.default.bind(_SlidePanelDocCommonTestModule.default);

var SlidePanelDemo = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(SlidePanelDemo, _React$Component);

  function SlidePanelDemo(props) {
    var _this;

    (0, _classCallCheck2.default)(this, SlidePanelDemo);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(SlidePanelDemo).call(this, props));
    _this.state = {
      panelIsOpen: false
    };
    _this.handlePanelToggle = _this.handlePanelToggle.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(SlidePanelDemo, [{
    key: "handlePanelToggle",
    value: function handlePanelToggle() {
      this.setState(function (prevState) {
        return {
          panelIsOpen: !prevState.panelIsOpen
        };
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", {
        className: cx('content-wrapper-toggle')
      }, _react.default.createElement(_SlidePanel.default, {
        id: "test-slide",
        mainContent: _react.default.createElement("div", {
          className: cx('main-content')
        }, _react.default.createElement("button", {
          type: "button",
          id: "test-toggle",
          className: cx('button'),
          onClick: this.handlePanelToggle
        }, "toggle")),
        panelContent: _react.default.createElement("div", {
          id: "panel-content",
          className: cx('panel-content')
        }, _react.default.createElement("button", {
          id: "focus-button",
          type: "button",
          className: cx('button'),
          onClick: this.handlePanelToggle
        }, "Close panel")),
        panelAriaLabel: "Panel content area",
        mainAriaLabel: "Main content area",
        panelSize: "small",
        panelBehavior: "overlay",
        isOpen: this.state.panelIsOpen,
        fill: true
      }));
    }
  }]);
  return SlidePanelDemo;
}(_react.default.Component);

var _default = SlidePanelDemo;
exports.default = _default;
