// terra-slide-panel/lib/terra-dev-site/test/slide-panel/SlidePanelFill.test.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _SlidePanel = _interopRequireWildcard(require("../../../SlidePanel"));

var _SlidePanelMainContent = _interopRequireDefault(require("../common/SlidePanelMainContent"));

var _SlidePanelPanelContent = _interopRequireDefault(require("../common/SlidePanelPanelContent"));

var _SlidePanelDocCommonTestModule = _interopRequireDefault(require("./SlidePanelDocCommon.test.module.scss"));

var cx = _bind.default.bind(_SlidePanelDocCommonTestModule.default);

var SlidePanelFill = function SlidePanelFill() {
  return _react.default.createElement("div", {
    className: cx('content-wrapper')
  }, _react.default.createElement(_SlidePanel.default, {
    mainContent: _react.default.createElement(_SlidePanelMainContent.default, null),
    panelContent: _react.default.createElement(_SlidePanelPanelContent.default, null),
    panelBehavior: "squish",
    panelPosition: _SlidePanel.SlidePanelPositions.END,
    panelSize: "small",
    isOpen: true,
    fill: true
  }));
};

var _default = SlidePanelFill;
exports.default = _default;
