// terra-slide-panel/lib/terra-dev-site/test/slide-panel/SlidePanelFullscreen.test.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SlidePanel = _interopRequireWildcard(require("../../../SlidePanel"));

var _SlidePanelMainContent = _interopRequireDefault(require("../common/SlidePanelMainContent"));

var _SlidePanelPanelContent = _interopRequireDefault(require("../common/SlidePanelPanelContent"));

var SlidePanelFullscreen = function SlidePanelFullscreen() {
  return _react.default.createElement(_SlidePanel.default, {
    mainContent: _react.default.createElement(_SlidePanelMainContent.default, null),
    panelContent: _react.default.createElement(_SlidePanelPanelContent.default, null),
    panelBehavior: "overlay",
    panelPosition: _SlidePanel.SlidePanelPositions.END,
    panelSize: "small",
    isOpen: true,
    isFullscreen: true
  });
};

var _default = SlidePanelFullscreen;
exports.default = _default;
