// terra-slide-panel/lib/terra-dev-site/test/slide-panel/SlidePanelSideToggle.test.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _SlidePanel = _interopRequireDefault(require("../../../SlidePanel"));

var _SlidePanelDocCommonTestModule = _interopRequireDefault(require("./SlidePanelDocCommon.test.module.scss"));

var cx = _bind.default.bind(_SlidePanelDocCommonTestModule.default);

var ButtonWrapper = function ButtonWrapper() {
  var _useState = (0, _react.useState)(0),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      count = _useState2[0],
      setCount = _useState2[1];

  var increaseCount = function increaseCount() {
    setCount(count + 1);
  };

  return _react.default.createElement("div", {
    id: "panel-content",
    className: cx('panel-content')
  }, _react.default.createElement("button", {
    id: "focus-button",
    type: "button",
    className: cx('button'),
    onClick: increaseCount
  }, "Increase Count "), count);
};

var SlidePanelSideDemo = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(SlidePanelSideDemo, _React$Component);

  function SlidePanelSideDemo(props) {
    var _this;

    (0, _classCallCheck2.default)(this, SlidePanelSideDemo);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(SlidePanelSideDemo).call(this, props));
    _this.state = {
      panelHasStartPostion: false
    };
    _this.handlePanelToggle = _this.handlePanelToggle.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(SlidePanelSideDemo, [{
    key: "handlePanelToggle",
    value: function handlePanelToggle() {
      this.setState(function (prevState) {
        return {
          panelHasStartPostion: !prevState.panelHasStartPostion
        };
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", {
        className: cx('content-wrapper-toggle')
      }, _react.default.createElement(_SlidePanel.default, {
        id: "test-slide",
        mainContent: _react.default.createElement("div", {
          className: cx('main-content')
        }, _react.default.createElement("button", {
          type: "button",
          id: "test-toggle",
          className: cx('button'),
          onClick: this.handlePanelToggle
        }, "toggle")),
        panelContent: _react.default.createElement(ButtonWrapper, null),
        panelAriaLabel: "Panel content area",
        mainAriaLabel: "Main content area",
        panelSize: "small",
        panelBehavior: "squish",
        panelPosition: this.state.panelHasStartPostion ? 'start' : 'end',
        isOpen: true,
        fill: true
      }));
    }
  }]);
  return SlidePanelSideDemo;
}(_react.default.Component);

var _default = SlidePanelSideDemo;
exports.default = _default;
