// terra-slide-panel/lib/terra-dev-site/test/slide-panel/SlidePanelSquishLarge.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _SlidePanel = _interopRequireDefault(require("../../../SlidePanel"));

var _SlidePanelDocCommonTestModule = _interopRequireDefault(require("./SlidePanelDocCommon.test.module.scss"));

var cx = _bind.default.bind(_SlidePanelDocCommonTestModule.default);

var SlidePanelSquishLarge = function SlidePanelSquishLarge() {
  return _react.default.createElement("div", {
    className: cx('content-wrapper-large')
  }, _react.default.createElement(_SlidePanel.default, {
    mainContent: _react.default.createElement("div", {
      className: cx('main-content')
    }),
    panelContent: _react.default.createElement("div", {
      className: cx('panel-content')
    }),
    panelBehavior: "squish",
    panelSize: "large",
    isOpen: true,
    fill: true
  }));
};

var _default = SlidePanelSquishLarge;
exports.default = _default;
