// terra-button/lib/terra-dev-site/doc/example/ButtonDisabled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconEdit = _interopRequireDefault(require("terra-icon/lib/icon/IconEdit"));

var _Button = _interopRequireDefault(require("terra-button/lib/Button"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _ButtonDocCommonModule = _interopRequireDefault(require("./ButtonDocCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_ButtonDocCommonModule.default);

var ButtonDisabled = function ButtonDisabled() {
  return _react.default.createElement("div", null, _react.default.createElement(_Button.default, {
    text: "Disabled Button",
    isDisabled: true,
    className: cx('button')
  }), _react.default.createElement(_Button.default, {
    text: "Disabled Emphasis",
    variant: "emphasis",
    isDisabled: true,
    className: cx('button')
  }), _react.default.createElement(_Button.default, {
    text: "Disabled Ghost",
    variant: "ghost",
    isDisabled: true,
    className: cx('button')
  }), _react.default.createElement(_Button.default, {
    text: "Disabled De-emphasis",
    variant: "de-emphasis",
    isDisabled: true,
    className: cx('button')
  }), _react.default.createElement(_Button.default, {
    text: "Disabled Action",
    variant: "action",
    isDisabled: true,
    icon: _react.default.createElement(_IconEdit.default, null),
    className: cx('button')
  }), _react.default.createElement(_Button.default, {
    text: "Disabled Utility",
    variant: "utility",
    isDisabled: true,
    icon: _react.default.createElement(_IconEdit.default, null),
    className: cx('button')
  }));
};

var _default = ButtonDisabled;
exports.default = _default;
