// terra-application/lib/terra-dev-site/test/application-loading-overlay/LoadingOverlayTest.test.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _ApplicationLoadingOverlay = _interopRequireDefault(require("../../../../lib/application-loading-overlay/ApplicationLoadingOverlay"));

var _ApplicationLoadingOverlayProvider = _interopRequireDefault(require("../../../../lib/application-loading-overlay/ApplicationLoadingOverlayProvider"));

var ApplicationLoadingOverlayTest = function ApplicationLoadingOverlayTest() {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      showClearOverlay = _useState2[0],
      setShowClearOverlay = _useState2[1];

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2.default)(_useState3, 2),
      showLightOverlay = _useState4[0],
      setShowLightOverlay = _useState4[1];

  var _useState5 = (0, _react.useState)(false),
      _useState6 = (0, _slicedToArray2.default)(_useState5, 2),
      showDarkOverlay = _useState6[0],
      setShowDarkOverlay = _useState6[1];

  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("button", {
    id: "reset-button",
    type: "button",
    onClick: function onClick() {
      setShowClearOverlay(false);
      setShowLightOverlay(false);
      setShowDarkOverlay(false);
    }
  }, "Reset"), _react.default.createElement("button", {
    id: "clear-button",
    type: "button",
    onClick: function onClick() {
      setShowClearOverlay(true);
    }
  }, "Show Clear Loading Overlay"), _react.default.createElement("button", {
    id: "light-button",
    type: "button",
    onClick: function onClick() {
      setShowLightOverlay(true);
    }
  }, "Show Light Loading Overlay"), _react.default.createElement("button", {
    id: "dark-button",
    type: "button",
    onClick: function onClick() {
      setShowDarkOverlay(true);
    }
  }, "Show Dark Loading Overlay"), _react.default.createElement(_ApplicationLoadingOverlayProvider.default, null, _react.default.createElement(_ApplicationLoadingOverlay.default, {
    isOpen: showClearOverlay
  }), _react.default.createElement(_ApplicationLoadingOverlay.default, {
    isOpen: showLightOverlay,
    backgroundStyle: "light"
  }), _react.default.createElement(_ApplicationLoadingOverlay.default, {
    isOpen: showDarkOverlay,
    backgroundStyle: "dark"
  })));
};

var _default = ApplicationLoadingOverlayTest;
exports.default = _default;
