// terra-form-select/lib/terra-dev-site/doc/example/legacy/Disabled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _terraFormSelect = _interopRequireDefault(require("terra-form-select"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _FormSelectDocCommonModule = _interopRequireDefault(require("../FormSelectDocCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_FormSelectDocCommonModule.default);

var DisabledExample = function DisabledExample() {
  return _react.default.createElement(_terraFormSelect.default, {
    placeholder: "Select a color",
    disabled: true,
    className: cx('form-select'),
    defaultValue: "blue"
  }, _react.default.createElement(_terraFormSelect.default.Option, {
    value: "blue",
    display: "Blue"
  }), _react.default.createElement(_terraFormSelect.default.Option, {
    value: "green",
    display: "Green"
  }), _react.default.createElement(_terraFormSelect.default.Option, {
    value: "purple",
    display: "Purple"
  }), _react.default.createElement(_terraFormSelect.default.Option, {
    value: "red",
    display: "Red"
  }), _react.default.createElement(_terraFormSelect.default.Option, {
    value: "violet",
    display: "Violet"
  }));
};

var _default = DisabledExample;
exports.default = _default;
