// terra-form-select/lib/terra-dev-site/doc/example/legacy/MaxSelection.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SelectField = _interopRequireDefault(require("terra-form-select/lib/SelectField"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _FormSelectDocCommonModule = _interopRequireDefault(require("../FormSelectDocCommon.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_FormSelectDocCommonModule.default);

var MaxSelectionExample = function MaxSelectionExample() {
  return _react.default.createElement(_SelectField.default, {
    help: "Shades of blue and green.",
    label: "Colors",
    maxSelectionCount: 3,
    placeholder: "Select a size",
    selectId: "max-selection-field",
    variant: "multiple",
    className: cx('form-select')
  }, _react.default.createElement(_SelectField.default.OptGroup, {
    label: "Shades of Blue"
  }, _react.default.createElement(_SelectField.default.Option, {
    value: "blue",
    display: "Blue"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "cyan",
    display: "Cyan"
  }), _react.default.createElement(_SelectField.default.Option, {
    disabled: true,
    value: "teal",
    display: "Teal"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "azul",
    display: "Azul"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "aqua",
    display: "Aqua"
  })), _react.default.createElement(_SelectField.default.OptGroup, {
    label: "Shades of Green"
  }, _react.default.createElement(_SelectField.default.Option, {
    value: "green",
    display: "Green"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "forest",
    display: "Forest Green"
  }), _react.default.createElement(_SelectField.default.Option, {
    disabled: true,
    value: "dark",
    display: "Dark Green"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "neon",
    display: "Neon Green"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "emerald",
    display: "Emerald"
  })));
};

var _default = MaxSelectionExample;
exports.default = _default;
