// terra-form-select/lib/terra-dev-site/test/form-select/IncompleteSelectField.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _SelectField = _interopRequireDefault(require("../../../SelectField"));

var _SelectTestModule = _interopRequireDefault(require("./common/Select.test.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_SelectTestModule.default);

var IncompleteSelectFieldExample = function IncompleteSelectFieldExample() {
  return _react.default.createElement("div", {
    className: cx('content-wrapper')
  }, _react.default.createElement(_SelectField.default, {
    required: true,
    isIncomplete: true,
    label: "T-shirt size",
    placeholder: "Select a size",
    selectId: "select-field",
    defaultValue: "small"
  }, _react.default.createElement(_SelectField.default.Option, {
    value: "small",
    display: "Small"
  })));
};

var _default = IncompleteSelectFieldExample;
exports.default = _default;
