// terra-form-select/lib/terra-dev-site/test/form-select/IsTouchAccessibleSearchSelect.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _SearchSelect = _interopRequireDefault(require("../../../SearchSelect"));

var _SelectTestModule = _interopRequireDefault(require("./common/Select.test.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_SelectTestModule.default);

var IsTouchAccessibleSearchSelect = function IsTouchAccessibleSearchSelect() {
  return _react.default.createElement("div", {
    className: cx('content-wrapper')
  }, _react.default.createElement(_SearchSelect.default, {
    placeholder: "Select a color",
    id: "search",
    isTouchAccessible: true
  }, _react.default.createElement(_SearchSelect.default.Option, {
    value: "blue",
    display: "Blue"
  }), _react.default.createElement(_SearchSelect.default.Option, {
    value: "green",
    display: "Green"
  }), _react.default.createElement(_SearchSelect.default.Option, {
    value: "purple",
    display: "Purple"
  }), _react.default.createElement(_SearchSelect.default.Option, {
    value: "red",
    display: "Red"
  }), _react.default.createElement(_SearchSelect.default.Option, {
    value: "violet",
    display: "Violet"
  })), _react.default.createElement("p", null, "Anim ad nulla est ipsum mollit mollit cillum consectetur."));
};

var _default = IsTouchAccessibleSearchSelect;
exports.default = _default;
