// terra-form-select/lib/terra-dev-site/test/form-select/MaxHeight.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Select = _interopRequireDefault(require("../../../Select"));

var _SelectTestModule = _interopRequireDefault(require("./common/Select.test.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_SelectTestModule.default);

var MaxHeight = function MaxHeight() {
  return _react.default.createElement("div", {
    className: cx('content-wrapper')
  }, _react.default.createElement(_Select.default, {
    placeholder: "Select a number",
    id: "maxHeight",
    defaultValue: "1",
    maxHeight: 300
  }, _react.default.createElement(_Select.default.Option, {
    value: "1",
    display: "One"
  }), _react.default.createElement(_Select.default.Option, {
    value: "2",
    display: "Two"
  }), _react.default.createElement(_Select.default.Option, {
    value: "3",
    display: "Three"
  }), _react.default.createElement(_Select.default.Option, {
    value: "4",
    display: "Four"
  }), _react.default.createElement(_Select.default.Option, {
    value: "5",
    display: "Five"
  }), _react.default.createElement(_Select.default.Option, {
    value: "6",
    display: "Six"
  }), _react.default.createElement(_Select.default.Option, {
    value: "7",
    display: "Seven"
  }), _react.default.createElement(_Select.default.Option, {
    value: "8",
    display: "Eight"
  }), _react.default.createElement(_Select.default.Option, {
    value: "9",
    display: "Nine"
  }), _react.default.createElement(_Select.default.Option, {
    value: "10",
    display: "Ten"
  }), _react.default.createElement(_Select.default.Option, {
    value: "11",
    display: "Eleven"
  }), _react.default.createElement(_Select.default.Option, {
    value: "12",
    display: "Twelve"
  }), _react.default.createElement(_Select.default.Option, {
    value: "13",
    display: "Thirteen"
  })));
};

var _default = MaxHeight;
exports.default = _default;
