// terra-form-select/lib/terra-dev-site/test/form-select/MaxSelection.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _SelectField = _interopRequireDefault(require("../../../SelectField"));

var _SelectTestModule = _interopRequireDefault(require("./common/Select.test.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_SelectTestModule.default);

var MaxSelectionExample = function MaxSelectionExample() {
  return _react.default.createElement("div", {
    className: cx('content-wrapper')
  }, _react.default.createElement(_SelectField.default, {
    id: "tag",
    label: "Colors",
    maxSelectionCount: 2,
    placeholder: "Select a color",
    selectId: "select-field",
    variant: "tag",
    help: "Shades of blue and green."
  }, _react.default.createElement(_SelectField.default.OptGroup, {
    label: "Shades of Blue"
  }, _react.default.createElement(_SelectField.default.Option, {
    value: "blue",
    display: "Blue"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "cyan",
    display: "Cyan"
  }), _react.default.createElement(_SelectField.default.Option, {
    disabled: true,
    value: "teal",
    display: "Teal"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "azul",
    display: "Azul"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "aqua",
    display: "Aqua"
  })), _react.default.createElement(_SelectField.default.OptGroup, {
    label: "Shades of Green"
  }, _react.default.createElement(_SelectField.default.Option, {
    value: "green",
    display: "Green"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "forest",
    display: "Forest Green"
  }), _react.default.createElement(_SelectField.default.Option, {
    disabled: true,
    value: "dark",
    display: "Dark Green"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "neon",
    display: "Neon Green"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "emerald",
    display: "Emerald"
  }))));
};

var _default = MaxSelectionExample;
exports.default = _default;
