// terra-form-select/lib/terra-dev-site/test/form-select/SelectField.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _SelectField = _interopRequireDefault(require("../../../SelectField"));

var _SelectTestModule = _interopRequireDefault(require("./common/Select.test.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_SelectTestModule.default);

var SelectFieldExample = function SelectFieldExample() {
  return _react.default.createElement("div", {
    className: cx('content-wrapper')
  }, _react.default.createElement(_SelectField.default, {
    label: "T-shirt size",
    placeholder: "Select a size",
    selectId: "select-field",
    defaultValue: "small"
  }, _react.default.createElement(_SelectField.default.Option, {
    value: "xSmall",
    display: "Extra Small"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "small",
    display: "Small"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "medium",
    display: "Medium"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "large",
    display: "Large"
  }), _react.default.createElement(_SelectField.default.Option, {
    value: "xLarge",
    display: "Extra Large"
  })));
};

var _default = SelectFieldExample;
exports.default = _default;
