// terra-form-select/lib/terra-dev-site/test/form-select/UncontrolledDefaultLongText.test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _Select = _interopRequireDefault(require("../../../Select"));

var _SelectTestModule = _interopRequireDefault(require("./common/Select.test.module.scss"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind.default.bind(_SelectTestModule.default);

var UncontrolledDefault = function UncontrolledDefault() {
  return _react.default.createElement("div", {
    className: cx('content-wrapper')
  }, _react.default.createElement(_Select.default, {
    placeholder: "Select a color",
    id: "default",
    defaultValue: "lorem"
  }, _react.default.createElement(_Select.default.Option, {
    value: "lorem",
    display: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor"
  }), _react.default.createElement(_Select.default.Option, {
    value: "blue",
    display: "Blue"
  }), _react.default.createElement(_Select.default.Option, {
    value: "purple",
    display: "Purple"
  }), _react.default.createElement(_Select.default.Option, {
    value: "red",
    display: "Red"
  }), _react.default.createElement(_Select.default.Option, {
    value: "violet",
    display: "Violet"
  })));
};

var _default = UncontrolledDefault;
exports.default = _default;
