// terra-abstract-modal/lib/terra-dev-site/test/abstract-modal/AbstractModalDisableCloseOnEsc.test.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _AbstractModal = _interopRequireDefault(require("../../../AbstractModal"));

require("./AbstractModalTestStyles.module.scss");

var ModalDisableCloseOnEsc = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(ModalDisableCloseOnEsc, _React$Component);

  function ModalDisableCloseOnEsc() {
    var _this;

    (0, _classCallCheck2.default)(this, ModalDisableCloseOnEsc);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(ModalDisableCloseOnEsc).call(this));
    _this.state = {
      isOpen: true
    };
    _this.handleOpenModal = _this.handleOpenModal.bind((0, _assertThisInitialized2.default)(_this));
    _this.handleCloseModal = _this.handleCloseModal.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(ModalDisableCloseOnEsc, [{
    key: "handleOpenModal",
    value: function handleOpenModal() {
      this.setState({
        isOpen: true
      });
    }
  }, {
    key: "handleCloseModal",
    value: function handleCloseModal() {
      this.setState({
        isOpen: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement(_AbstractModal.default, {
        ariaLabel: "Terra Modal",
        isOpen: this.state.isOpen,
        onRequestClose: this.handleCloseModal,
        closeOnEsc: false,
        classNameModal: "test-background-class"
      }, _react.default.createElement("div", null, _react.default.createElement("h1", null, "Terra Modal"), _react.default.createElement("h2", null, "Subtitle"), _react.default.createElement("hr", null), _react.default.createElement("p", null, "The Terra Modal is appended to the document body."), _react.default.createElement("p", null, 'Modal is assigned a role of \'document\' for accessibility.'), _react.default.createElement("button", {
        id: "focus-button",
        type: "button",
        onClick: this.handleCloseModal
      }, "Close Modal"))), _react.default.createElement("button", {
        type: "button",
        className: "button-open-modal",
        onClick: this.handleOpenModal
      }, "Open Modal"));
    }
  }]);
  return ModalDisableCloseOnEsc;
}(_react.default.Component);

var _default = ModalDisableCloseOnEsc;
exports.default = _default;
