// terra-theme-provider/lib/terra-dev-site/doc/example/DefaultThemeProvider.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _bind = _interopRequireDefault(require("classnames/bind"));

var _terraThemeProvider = _interopRequireDefault(require("terra-theme-provider"));

var _MockThemeComponent = _interopRequireDefault(require("terra-theme-provider/lib/terra-dev-site/doc/example/MockThemeComponent"));

var _DefaultThemeProviderModule = _interopRequireDefault(require("./DefaultThemeProvider.module.scss"));

var cx = _bind.default.bind(_DefaultThemeProviderModule.default);

var DefaultThemeProvider = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(DefaultThemeProvider, _React$Component);

  function DefaultThemeProvider(props) {
    var _this;

    (0, _classCallCheck2.default)(this, DefaultThemeProvider);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(DefaultThemeProvider).call(this, props));
    _this.state = {
      theme: ''
    };
    _this.handleThemeChange = _this.handleThemeChange.bind((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(DefaultThemeProvider, [{
    key: "handleThemeChange",
    value: function handleThemeChange(e) {
      this.setState({
        theme: e.target.value
      });
    }
  }, {
    key: "render",
    value: function render() {
      var themeSwitcher;

      function supportsCSSVars() {
        return window.CSS && window.CSS.supports && window.CSS.supports('(--fake-var: 0)');
      }

      if (supportsCSSVars()) {
        themeSwitcher = _react.default.createElement("div", {
          className: cx('theme-switcher-wrapper')
        }, _react.default.createElement("label", {
          htmlFor: "theme"
        }, " Theme: "), _react.default.createElement("select", {
          id: "theme",
          value: this.state.theme,
          onChange: this.handleThemeChange
        }, _react.default.createElement("option", {
          value: ""
        }, "Default"), _react.default.createElement("option", {
          value: _terraThemeProvider.default.Opts.Themes.MOCK
        }, "Mock Theme")));
      } else {
        themeSwitcher = _react.default.createElement("div", null);
      }

      return _react.default.createElement("div", null, _react.default.createElement("div", null, themeSwitcher), _react.default.createElement(_terraThemeProvider.default, {
        themeName: this.state.theme
      }, _react.default.createElement("div", null, _react.default.createElement(_MockThemeComponent.default, null, "Themable component"))));
    }
  }]);
  return DefaultThemeProvider;
}(_react.default.Component);

var _default = DefaultThemeProvider;
exports.default = _default;
