// terra-i18n/lib/terra-dev-site/doc/example/I18nDemo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactIntl = require("react-intl");

var _terraI18n = require("terra-i18n");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var Demo = /*#__PURE__*/function (_React$Component) {
  _inherits(Demo, _React$Component);

  function Demo(props) {
    var _this;

    _classCallCheck(this, Demo);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Demo).call(this, props));
    _this.state = {
      areTranslationsLoaded: false,
      locale: 'en-US',
      messages: {}
    };
    _this.handleLocaleChange = _this.handleLocaleChange.bind(_assertThisInitialized(_this));
    return _this;
  }

  _createClass(Demo, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      (0, _terraI18n.i18nLoader)(this.state.locale, this.setState, this);
    }
  }, {
    key: "handleLocaleChange",
    value: function handleLocaleChange(e) {
      (0, _terraI18n.i18nLoader)(e.target.value, this.setState, this);
    }
  }, {
    key: "render",
    value: function render() {
      if (!this.state.areTranslationsLoaded) {
        return _react.default.createElement("div", null);
      }

      return _react.default.createElement(_terraI18n.I18nProvider, {
        locale: this.state.locale,
        messages: this.state.messages
      }, _react.default.createElement("div", null, _react.default.createElement("span", null, " Example Message Format: "), _react.default.createElement(_reactIntl.FormattedMessage, {
        id: "Terra.ajax.error"
      })), _react.default.createElement("p", null, _react.default.createElement("span", null, " Example Number Format: "), _react.default.createElement(_reactIntl.FormattedNumber, {
        value: parseFloat('1123432.123')
      })), _react.default.createElement("p", null, _react.default.createElement("span", null, "Example Date Format: "), _react.default.createElement(_reactIntl.FormattedDate, {
        value: new Date(1575476163287)
      })), _react.default.createElement("label", {
        htmlFor: "locale"
      }, ' ', "Current locale:", this.state.locale, ' '), _react.default.createElement("select", {
        id: "locale",
        value: this.state.locale,
        onChange: this.handleLocaleChange
      }, _react.default.createElement("option", {
        value: "en"
      }, "en"), _react.default.createElement("option", {
        value: "en-GB"
      }, "en-GB"), _react.default.createElement("option", {
        value: "en-US"
      }, "en-US"), _react.default.createElement("option", {
        value: "de"
      }, "de"), _react.default.createElement("option", {
        value: "es"
      }, "es"), _react.default.createElement("option", {
        value: "fr"
      }, "fr"), _react.default.createElement("option", {
        value: "nl"
      }, "nl"), _react.default.createElement("option", {
        value: "pt"
      }, "pt")));
    }
  }]);

  return Demo;
}(_react.default.Component);

var _default = function _default() {
  return _react.default.createElement(Demo, null);
};

exports.default = _default;
